<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Introduction to Data Science - 4008Y</title>
  <meta name="description" content="A brief introduction so some fundamental applications of R in data science" />
  <meta name="generator" content="bookdown #bookdown:version# and GitBook 2.6.7" />

  <meta property="og:title" content="Introduction to Data Science - 4008Y" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A brief introduction so some fundamental applications of R in data science" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Introduction to Data Science - 4008Y" />
  
  <meta name="twitter:description" content="A brief introduction so some fundamental applications of R in data science" />
  

<meta name="author" content="Dr Ellen Bell" />


<meta name="date" content="2023-02-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<!--bookdown:link_prev-->
<!--bookdown:link_next-->
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="libs/klippy-0.0.0.9500/js/klippy.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



<!--bookdown:title:start-->
<div id="header">
<h1 class="title">Introduction to Data Science - 4008Y</h1>
<p class="author"><em>Dr Ellen Bell</em></p>
<p class="date"><em>2023-02-15</em></p>
</div>
<!--bookdown:title:end-->

<!--bookdown:toc:start-->
  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">
<!--bookdown:toc2:start-->
<ul>
<li><a href="#welcome-to-the-data-science-workbook" id="toc-welcome-to-the-data-science-workbook"><span class="toc-section-number">1</span> Welcome to the Data Science Workbook<span></span></a>
<ul>
<li><a href="#data-science-learning-objectives" id="toc-data-science-learning-objectives"><span class="toc-section-number">1.1</span> Data Science learning objectives<span></span></a></li>
<li><a href="#teaching-layout-for-data-science" id="toc-teaching-layout-for-data-science"><span class="toc-section-number">1.2</span> Teaching layout for Data Science<span></span></a></li>
<li><a href="#assessment" id="toc-assessment"><span class="toc-section-number">1.3</span> Assessment<span></span></a></li>
<li><a href="#how-to-use-this-workbook" id="toc-how-to-use-this-workbook"><span class="toc-section-number">1.4</span> How to use this workbook<span></span></a></li>
<li><a href="#why-is-data-science-important-and-why-are-we-teaching-it-in-r" id="toc-why-is-data-science-important-and-why-are-we-teaching-it-in-r"><span class="toc-section-number">1.5</span> Why is data science important and why are we teaching it in R?<span></span></a></li>
</ul></li>
<li><a href="#introduction-to-r-and-posit-cloud---week-2a" id="toc-introduction-to-r-and-posit-cloud---week-2a"><span class="toc-section-number">2</span> Introduction to R and posit Cloud - Week 2A<span></span></a>
<ul>
<li><a href="#what-is-r-and-posit-cloud" id="toc-what-is-r-and-posit-cloud"><span class="toc-section-number">2.1</span> What is R and posit Cloud?<span></span></a></li>
<li><a href="#a-warning-on-managing-your-work-flow" id="toc-a-warning-on-managing-your-work-flow"><span class="toc-section-number">2.2</span> A warning on managing your work flow<span></span></a></li>
<li><a href="#creating-your-first-posit-cloud-project" id="toc-creating-your-first-posit-cloud-project"><span class="toc-section-number">2.3</span> Creating your first posit Cloud Project<span></span></a></li>
<li><a href="#maximising-the-efficiency-of-your-workspace" id="toc-maximising-the-efficiency-of-your-workspace"><span class="toc-section-number">2.4</span> Maximising the efficiency of your workspace<span></span></a></li>
<li><a href="#entering-commands-directly-into-the-consol" id="toc-entering-commands-directly-into-the-consol"><span class="toc-section-number">2.5</span> Entering commands directly into the consol<span></span></a></li>
<li><a href="#sending-commands-down-to-the-console-from-an-r-script" id="toc-sending-commands-down-to-the-console-from-an-r-script"><span class="toc-section-number">2.6</span> Sending commands down to the console from an R script<span></span></a></li>
<li><a href="#a-note-on-comments" id="toc-a-note-on-comments"><span class="toc-section-number">2.7</span> A note on comments<span></span></a></li>
</ul></li>
<li><a href="#objects-functions-and-packages---week-3a" id="toc-objects-functions-and-packages---week-3a"><span class="toc-section-number">3</span> Objects, Functions and Packages - Week 3A<span></span></a>
<ul>
<li><a href="#creating-objects" id="toc-creating-objects"><span class="toc-section-number">3.1</span> Creating objects<span></span></a></li>
<li><a href="#what-are-functions-and-packages" id="toc-what-are-functions-and-packages"><span class="toc-section-number">3.2</span> What are functions and packages?<span></span></a>
<ul>
<li><a href="#using-functions" id="toc-using-functions"><span class="toc-section-number">3.2.1</span> Using functions<span></span></a></li>
</ul></li>
<li><a href="#installing-packages" id="toc-installing-packages"><span class="toc-section-number">3.3</span> Installing packages<span></span></a></li>
<li><a href="#what-do-i-do-if-i-get-an-error-message" id="toc-what-do-i-do-if-i-get-an-error-message"><span class="toc-section-number">3.4</span> What do I do if I get an error message?<span></span></a></li>
<li><a href="#references" id="toc-references"><span class="toc-section-number">3.5</span> References<span></span></a></li>
</ul></li>
<li><a href="#directories-and-data-frames---week-4a---workshop-1" id="toc-directories-and-data-frames---week-4a---workshop-1"><span class="toc-section-number">4</span> Directories and data frames - Week 4A - Workshop 1<span></span></a>
<ul>
<li><a href="#data-entry" id="toc-data-entry"><span class="toc-section-number">4.1</span> Data Entry<span></span></a>
<ul>
<li><a href="#a-note-on-naming" id="toc-a-note-on-naming"><span class="toc-section-number">4.1.1</span> A note on naming<span></span></a></li>
</ul></li>
<li><a href="#setting-up-your-r-workspace" id="toc-setting-up-your-r-workspace"><span class="toc-section-number">4.2</span> Setting up your R workspace<span></span></a>
<ul>
<li><a href="#a-note-on-directories-and-their-structure" id="toc-a-note-on-directories-and-their-structure"><span class="toc-section-number">4.2.1</span> A note on directories and their structure<span></span></a></li>
</ul></li>
<li><a href="#setting-up-your-script" id="toc-setting-up-your-script"><span class="toc-section-number">4.3</span> Setting up your script<span></span></a></li>
<li><a href="#importing-data" id="toc-importing-data"><span class="toc-section-number">4.4</span> Importing data<span></span></a></li>
<li><a href="#creating-tibbles" id="toc-creating-tibbles"><span class="toc-section-number">4.5</span> Creating tibbles<span></span></a></li>
<li><a href="#before-you-leave" id="toc-before-you-leave"><span class="toc-section-number">4.6</span> Before you leave!<span></span></a></li>
<li><a href="#references-1" id="toc-references-1"><span class="toc-section-number">4.7</span> References<span></span></a></li>
</ul></li>
<li><a href="#reading-week---week-5a" id="toc-reading-week---week-5a"><span class="toc-section-number">5</span> Reading week - Week 5A<span></span></a></li>
<li><a href="#exploring-the-dimensions-of-your-data-frame---week-6a" id="toc-exploring-the-dimensions-of-your-data-frame---week-6a"><span class="toc-section-number">6</span> Exploring the dimensions of your data frame - Week 6A<span></span></a>
<ul>
<li><a href="#top-to-tail---some-useful-functions-to-explore" id="toc-top-to-tail---some-useful-functions-to-explore"><span class="toc-section-number">6.1</span> Top to tail - some useful functions to explore<span></span></a></li>
<li><a href="#whats-your-sample-size" id="toc-whats-your-sample-size"><span class="toc-section-number">6.2</span> Whats your sample size?<span></span></a></li>
<li><a href="#before-you-leave-1" id="toc-before-you-leave-1"><span class="toc-section-number">6.3</span> Before you leave!<span></span></a></li>
<li><a href="#references-2" id="toc-references-2"><span class="toc-section-number">6.4</span> References<span></span></a></li>
</ul></li>
<li><a href="#a-plot-to-choose-from---week-7a" id="toc-a-plot-to-choose-from---week-7a"><span class="toc-section-number">7</span> A plot to choose from - Week 7A<span></span></a>
<ul>
<li><a href="#relationships-differences-and-distributions" id="toc-relationships-differences-and-distributions"><span class="toc-section-number">7.1</span> Relationships, differences and distributions<span></span></a></li>
<li><a href="#my-first-histogram-in-r" id="toc-my-first-histogram-in-r"><span class="toc-section-number">7.2</span> My first histogram in R<span></span></a></li>
<li><a href="#exporting-your-histogram" id="toc-exporting-your-histogram"><span class="toc-section-number">7.3</span> Exporting your histogram<span></span></a></li>
<li><a href="#before-you-leave-2" id="toc-before-you-leave-2"><span class="toc-section-number">7.4</span> Before you leave!<span></span></a></li>
<li><a href="#references-3" id="toc-references-3"><span class="toc-section-number">7.5</span> References<span></span></a></li>
</ul></li>
<li><a href="#lost-the-plot-some-more-data-visualisations---week-8a---workshop-2" id="toc-lost-the-plot-some-more-data-visualisations---week-8a---workshop-2"><span class="toc-section-number">8</span> Lost the plot? Some more data visualisations - Week 8A - Workshop 2<span></span></a>
<ul>
<li><a href="#looking-at-differences" id="toc-looking-at-differences"><span class="toc-section-number">8.1</span> Looking at differences<span></span></a></li>
<li><a href="#looking-at-relationships" id="toc-looking-at-relationships"><span class="toc-section-number">8.2</span> Looking at relationships<span></span></a></li>
<li><a href="#before-you-leave-3" id="toc-before-you-leave-3"><span class="toc-section-number">8.3</span> Before you leave!<span></span></a></li>
<li><a href="#references-4" id="toc-references-4"><span class="toc-section-number">8.4</span> References<span></span></a></li>
</ul></li>
<li><a href="#make-them-pretty---week-9a" id="toc-make-them-pretty---week-9a"><span class="toc-section-number">9</span> Make them pretty - Week 9A<span></span></a>
<ul>
<li><a href="#deconstructing-ggplot" id="toc-deconstructing-ggplot"><span class="toc-section-number">9.1</span> Deconstructing ggplot<span></span></a></li>
<li><a href="#labels" id="toc-labels"><span class="toc-section-number">9.2</span> Labels<span></span></a></li>
<li><a href="#switching-up-colours" id="toc-switching-up-colours"><span class="toc-section-number">9.3</span> Switching up colours<span></span></a></li>
<li><a href="#before-you-leave-4" id="toc-before-you-leave-4"><span class="toc-section-number">9.4</span> Before you leave!<span></span></a></li>
<li><a href="#references-5" id="toc-references-5"><span class="toc-section-number">9.5</span> References<span></span></a></li>
</ul></li>
<li><a href="#make-them-pretty-continued---week-10a" id="toc-make-them-pretty-continued---week-10a"><span class="toc-section-number">10</span> Make them pretty continued - Week 10A<span></span></a>
<ul>
<li><a href="#spacing" id="toc-spacing"><span class="toc-section-number">10.1</span> Spacing<span></span></a></li>
<li><a href="#themes" id="toc-themes"><span class="toc-section-number">10.2</span> Themes<span></span></a></li>
<li><a href="#before-you-leave-5" id="toc-before-you-leave-5"><span class="toc-section-number">10.3</span> Before you leave!<span></span></a></li>
<li><a href="#references-6" id="toc-references-6"><span class="toc-section-number">10.4</span> References<span></span></a></li>
</ul></li>
<li><a href="#the-subtle-art-of-cannibalisation---week-11a---workshop-3" id="toc-the-subtle-art-of-cannibalisation---week-11a---workshop-3"><span class="toc-section-number">11</span> The subtle art of cannibalisation - Week 11A - Workshop 3<span></span></a>
<ul>
<li><a href="#starting-on-the-right-footing" id="toc-starting-on-the-right-footing"><span class="toc-section-number">11.1</span> Starting on the right footing<span></span></a></li>
<li><a href="#beautifying-your-box-plots" id="toc-beautifying-your-box-plots"><span class="toc-section-number">11.2</span> Beautifying your box plots<span></span></a></li>
<li><a href="#beautifying-your-histograms" id="toc-beautifying-your-histograms"><span class="toc-section-number">11.3</span> Beautifying your histograms<span></span></a></li>
<li><a href="#beautifying-your-scatter-plot" id="toc-beautifying-your-scatter-plot"><span class="toc-section-number">11.4</span> Beautifying your scatter plot<span></span></a></li>
<li><a href="#a-challenge" id="toc-a-challenge"><span class="toc-section-number">11.5</span> A Challenge!<span></span></a></li>
<li><a href="#end-of-semester-1" id="toc-end-of-semester-1"><span class="toc-section-number">11.6</span> End of Semester 1<span></span></a></li>
<li><a href="#before-you-leave-6" id="toc-before-you-leave-6"><span class="toc-section-number">11.7</span> Before you leave!<span></span></a></li>
<li><a href="#references-7" id="toc-references-7"><span class="toc-section-number">11.8</span> References<span></span></a></li>
</ul></li>
<li><a href="#merry-christmas" id="toc-merry-christmas"><span class="toc-section-number">12</span> Merry Christmas!!!<span></span></a></li>
<li><a href="#new-year-new-data---workshop-4---athletes-week-2b" id="toc-new-year-new-data---workshop-4---athletes-week-2b"><span class="toc-section-number">13</span> New year, new data - Workshop 4 - Athletes, Week 2B<span></span></a>
<ul>
<li><a href="#a-new-project" id="toc-a-new-project"><span class="toc-section-number">13.1</span> A new project<span></span></a>
<ul>
<li><a href="#project-setup" id="toc-project-setup"><span class="toc-section-number">13.1.1</span> Project setup<span></span></a></li>
<li><a href="#script-set-up" id="toc-script-set-up"><span class="toc-section-number">13.1.2</span> Script set up<span></span></a></li>
<li><a href="#the-data" id="toc-the-data"><span class="toc-section-number">13.1.3</span> The data<span></span></a></li>
</ul></li>
<li><a href="#check-the-data" id="toc-check-the-data"><span class="toc-section-number">13.2</span> Check the data<span></span></a>
<ul>
<li><a href="#task-1---make-sure-your-data-are-clean-and-tidy" id="toc-task-1---make-sure-your-data-are-clean-and-tidy"><span class="toc-section-number">13.2.1</span> Task 1 - Make sure your data are clean and tidy<span></span></a></li>
</ul></li>
<li><a href="#explore-the-data" id="toc-explore-the-data"><span class="toc-section-number">13.3</span> Explore the data<span></span></a>
<ul>
<li><a href="#task-2---look-at-your-data-distributions" id="toc-task-2---look-at-your-data-distributions"><span class="toc-section-number">13.3.1</span> Task 2 - Look at your data distributions<span></span></a></li>
<li><a href="#task-3---look-for-differences-and-relationships" id="toc-task-3---look-for-differences-and-relationships"><span class="toc-section-number">13.3.2</span> Task 3 - Look for differences and relationships<span></span></a></li>
</ul></li>
<li><a href="#wrapping-up" id="toc-wrapping-up"><span class="toc-section-number">13.4</span> Wrapping up<span></span></a></li>
<li><a href="#before-you-leave-7" id="toc-before-you-leave-7"><span class="toc-section-number">13.5</span> Before you leave!<span></span></a></li>
<li><a href="#references-8" id="toc-references-8"><span class="toc-section-number">13.6</span> References<span></span></a></li>
</ul></li>
<li><a href="#playing-with-some-statistics---workshop-5---athletes-continued-week-4b" id="toc-playing-with-some-statistics---workshop-5---athletes-continued-week-4b"><span class="toc-section-number">14</span> Playing with some statistics - Workshop 5 - Athletes continued, Week 4B<span></span></a>
<ul>
<li><a href="#descriptive-statistics" id="toc-descriptive-statistics"><span class="toc-section-number">14.1</span> Descriptive statistics<span></span></a>
<ul>
<li><a href="#task-1---calculating-some-descriptive-statistics" id="toc-task-1---calculating-some-descriptive-statistics"><span class="toc-section-number">14.1.1</span> Task 1 - Calculating some descriptive statistics<span></span></a></li>
</ul></li>
<li><a href="#introducing-some-inferential-statistics" id="toc-introducing-some-inferential-statistics"><span class="toc-section-number">14.2</span> Introducing some inferential statistics<span></span></a>
<ul>
<li><a href="#the-general-linear-model" id="toc-the-general-linear-model"><span class="toc-section-number">14.2.1</span> The general linear model<span></span></a></li>
<li><a href="#task-2---testing-for-differences" id="toc-task-2---testing-for-differences"><span class="toc-section-number">14.2.2</span> Task 2 - Testing for differences<span></span></a></li>
<li><a href="#task-3---testing-for-relationships" id="toc-task-3---testing-for-relationships"><span class="toc-section-number">14.2.3</span> Task 3 - Testing for relationships<span></span></a></li>
</ul></li>
<li><a href="#assumptions" id="toc-assumptions"><span class="toc-section-number">14.3</span> Assumptions<span></span></a>
<ul>
<li><a href="#task-4---checking-for-normality" id="toc-task-4---checking-for-normality"><span class="toc-section-number">14.3.1</span> Task 4 - Checking for normality<span></span></a></li>
<li><a href="#task-5---what-to-do-if-your-data-are-not-normally-distributed" id="toc-task-5---what-to-do-if-your-data-are-not-normally-distributed"><span class="toc-section-number">14.3.2</span> Task 5 - What to do if your data are not normally distributed?<span></span></a></li>
</ul></li>
<li><a href="#wrapping-up-1" id="toc-wrapping-up-1"><span class="toc-section-number">14.4</span> Wrapping up<span></span></a></li>
<li><a href="#before-you-leave-8" id="toc-before-you-leave-8"><span class="toc-section-number">14.5</span> Before you leave!<span></span></a></li>
<li><a href="#references-9" id="toc-references-9"><span class="toc-section-number">14.6</span> References<span></span></a></li>
</ul></li>
<li><a href="#reporting-your-analysis---workshop-6---athletes-continued-week-6b" id="toc-reporting-your-analysis---workshop-6---athletes-continued-week-6b"><span class="toc-section-number">15</span> Reporting your analysis - Workshop 6 - Athletes continued, Week 6B<span></span></a>
<ul>
<li><a href="#selecting-what-to-report" id="toc-selecting-what-to-report"><span class="toc-section-number">15.1</span> Selecting what to report<span></span></a>
<ul>
<li><a href="#task-1---start-thinking-about-how-you-might-report-your-data" id="toc-task-1---start-thinking-about-how-you-might-report-your-data"><span class="toc-section-number">15.1.1</span> Task 1 - Start thinking about how you might report your data<span></span></a></li>
<li><a href="#task-2---building-your-first-multi-pannel-figure" id="toc-task-2---building-your-first-multi-pannel-figure"><span class="toc-section-number">15.1.2</span> Task 2 - Building your first multi-pannel figure<span></span></a></li>
</ul></li>
<li><a href="#writing-about-your-analysis" id="toc-writing-about-your-analysis"><span class="toc-section-number">15.2</span> Writing about your analysis<span></span></a>
<ul>
<li><a href="#task-3---writing-a-figure-legend" id="toc-task-3---writing-a-figure-legend"><span class="toc-section-number">15.2.1</span> Task 3 - Writing a figure legend<span></span></a></li>
<li><a href="#task-4---have-a-go-at-preparing-a-result-section" id="toc-task-4---have-a-go-at-preparing-a-result-section"><span class="toc-section-number">15.2.2</span> Task 4 - Have a go at preparing a result section<span></span></a></li>
</ul></li>
<li><a href="#wrapping-up-2" id="toc-wrapping-up-2"><span class="toc-section-number">15.3</span> Wrapping up<span></span></a></li>
<li><a href="#before-you-leave-9" id="toc-before-you-leave-9"><span class="toc-section-number">15.4</span> Before you leave!<span></span></a></li>
<li><a href="#references-10" id="toc-references-10"><span class="toc-section-number">15.5</span> References<span></span></a></li>
</ul></li>
<li><a href="#portfolio-tasks" id="toc-portfolio-tasks"><span class="toc-section-number">16</span> Portfolio Tasks<span></span></a>
<ul>
<li><a href="#background" id="toc-background"><span class="toc-section-number">16.1</span> Background<span></span></a></li>
<li><a href="#the-data-1" id="toc-the-data-1"><span class="toc-section-number">16.2</span> The data<span></span></a></li>
<li><a href="#the-tasks" id="toc-the-tasks"><span class="toc-section-number">16.3</span> The tasks<span></span></a>
<ul>
<li><a href="#exercise-1---plots" id="toc-exercise-1---plots"><span class="toc-section-number">16.3.1</span> Exercise 1 - plots<span></span></a></li>
<li><a href="#exercise-2---figure-legend" id="toc-exercise-2---figure-legend"><span class="toc-section-number">16.3.2</span> Exercise 2 - figure legend<span></span></a></li>
<li><a href="#exercise-3---justification" id="toc-exercise-3---justification"><span class="toc-section-number">16.3.3</span> Exercise 3 - justification<span></span></a></li>
<li><a href="#exercise-4---proof-of-programing" id="toc-exercise-4---proof-of-programing"><span class="toc-section-number">16.3.4</span> Exercise 4 - proof of programing<span></span></a></li>
</ul></li>
<li><a href="#references-11" id="toc-references-11"><span class="toc-section-number">16.4</span> References<span></span></a></li>
</ul></li>
</ul>
<!--bookdown:toc2:end-->
      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to Data Science - 4008Y</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<!--bookdown:toc:end-->
<!--bookdown:body:start-->
<div id="welcome-to-the-data-science-workbook" class="section level1 hasAnchor" number="1">
<h1 class="hasAnchor"><span class="header-section-number">1</span> Welcome to the Data Science Workbook<a href="#welcome-to-the-data-science-workbook" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="data-science-learning-objectives" class="section level2 hasAnchor" number="1.1">
<h2 class="hasAnchor"><span class="header-section-number">1.1</span> Data Science learning objectives<a href="#data-science-learning-objectives" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Introduction to Data Science is a ‘sub-module’ that will run throughout your first year as part of the BIO-4008Y Skills for Biologists module. Here we aim to teach you the fundamentals of how to collect and organise data in a clean and sensible format and how to correctly visualise and describe data. The platform we will be using to manage, manipulate and analyse data is posit Cloud. This is a cloud based interface that utilises the programming language R in a very similar interface to RStudio (a package you may wish to install on your own computers at some point). Over the course of the next few weeks you will start to learn fundamentals of this language and its application in posit Cloud.</p>
</div>
<div id="teaching-layout-for-data-science" class="section level2 hasAnchor" number="1.2">
<h2 class="hasAnchor"><span class="header-section-number">1.2</span> Teaching layout for Data Science<a href="#teaching-layout-for-data-science" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In Semester 1 we will be learning some basic R syntax, how to create and load data sets and how to produce some basic data visualisations that are attractive and reproducible. Each week you will have either a lecture or a 1 hour workshop, you will also be expected to complete a chapter of this workbook each week, in your own time.</p>
<p>Teaching for semester 1;</p>
<ul>
<li>Semester 1 Lectures - Weeks; 2, 3, 6, 7, 9, 10</li>
<li>Semester 1 Workshops - Weeks; 4, 8, 11</li>
</ul>
<p>In semester 2 will move onto some slightly more advanced data interpretation and visualisation skills and some descriptive statistics. Towards the end of the module we will briefly touch on inferential statistics, but this will be a very brief introduction. You will have a lecture every week until week 8 and a 2 hour workshop every other week. The workshops are all outlined as chapters in this workbook, if you don’t complete a workshop in the timetabled slot, we strongly recommend you make every effort to complete it in your own time.</p>
<p>Teaching for semester 2;</p>
<ul>
<li>Semester 2 Lectures - Weeks; 1, 2, 3, 4, 5, 6, 7, 8</li>
<li>Semester 2 Workshops - Weeks; 2, 4, 6, 8</li>
</ul>
</div>
<div id="assessment" class="section level2 hasAnchor" number="1.3">
<h2 class="hasAnchor"><span class="header-section-number">1.3</span> Assessment<a href="#assessment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Semester 1</li>
</ul>
<p>Two quizzes are available on the 4008Y page under Data Science. These are “semi-formative” quizzes, you can attempt them at any time and as many times as you like before the summative essay deadline for Skills for Biologists, the highest mark you achieve will be the one you are awarded. If you complete them with a grade of 40% or higher you will gain a percentage point towards your summative Skills essay. If you complete them with a grade of 80% or higher you will get 2 percentage points added to your summative Skills essay. That’s 4 percent in total, last year this was enough to move several students up to the next grade boundary, definitely worth doing!</p>
<ul>
<li>Semester 2</li>
</ul>
<p>As part of your Skills portfolio, there will be two data handling tasks. You will be given some data and asked to correctly visualise it in a clean, clear and attractive way, with a suitable figure legend. You will also be asked to write a justification for the decisions you made while preparing your data and visualising it. You will learn all the skills you need to do this while completing lectures, workshops or chapters in this workbook.</p>
</div>
<div id="how-to-use-this-workbook" class="section level2 hasAnchor" number="1.4">
<h2 class="hasAnchor"><span class="header-section-number">1.4</span> How to use this workbook<a href="#how-to-use-this-workbook" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In semester 1 each week of taught material is accompanied by a workbook chapter. You will be expected read through these chapters and complete any exercises in your own time. Each chapter should take no more then an hour to complete and in many cases will take much less time. I will aim to set aside 10 minutes at the end of each lecture to cover any potential problems you may be having, so if you get stuck, use this time to ask for help. In semester 2 we have more face to face workshop time, in each workshop you will be expected to work through a chapter in this workbook, you may not complete this in the allocated time, but will be expected to do so in your own time. Use face to face time to ask lots of questions, especially if you are stuck.</p>
</div>
<div id="why-is-data-science-important-and-why-are-we-teaching-it-in-r" class="section level2 hasAnchor" number="1.5">
<h2 class="hasAnchor"><span class="header-section-number">1.5</span> Why is data science important and why are we teaching it in R?<a href="#why-is-data-science-important-and-why-are-we-teaching-it-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It is a common misconception, that to be a good biologist you need to ‘know’ the mechanics of how life works. Questions like; “How does a cell undergo respiration?”, “How do kidneys filter blood?”, “How do some plants fix nitrogen?” and “How do honey bees communicate the location and quantity of resources to each other?”, spring to mind. While these questions are important, they are also, now, fairly well understood. But how did biologists come to their understanding of these mechanisms? The answer to this lies in data.</p>
<p>Good science is based on empirical observations and these observations should be reliably collected and reproducible. Exploration of theories through observations and experimentation leads to the collection of data and analysis and interpretation of data feeds into the bedrock of our understanding of how life works, i.e. the biological sciences.</p>
<p>Hopefully you can see why data handling, analysis and interpretation are important. So why are we teaching you how to handle data using R?</p>
<p>To many of you the use of programming languages, such as R, will be new. However if you can get into the habit of manipulating and analysing data in R you will be well set on the path to becoming an efficient and effective data analyst. Being confident with data is a key skill in the sciences and will serve you well in many career paths. In addition, knowledge and experience of programming languages such as R are fast becoming key skills in their own right, in science, industry, government and beyond.</p>
<p>In terms of the use of R for data handling and analysis, you may notice that the term <strong>reproducible</strong> reoccurs throughout this workbook. In the same way that your methods of data collection should be conducted and recorded in such a way that they may be reproduced by others, your data manipulation and analysis must also be conducted and recorded so as to be reproducible. Using R within the posit Cloud interface (we will come back to this in Chapter 2) makes it easy to record your data manipulation and analysis workflow and, if done well, makes it very easy for others to see and repeat what you have done.</p>
<p>So hopefully I have convinced you that data handling, analysis and R are all worthwhile skills to cultivate. We will spend some time in lectures discussing this as well. But for now take a look at Chapter 2. Happy coding!</p>
<!--chapter:end:index.Rmd-->
</div>
</div>
<div id="introduction-to-r-and-posit-cloud---week-2a" class="section level1 hasAnchor" number="2">
<h1 class="hasAnchor"><span class="header-section-number">2</span> Introduction to R and posit Cloud - Week 2A<a href="#introduction-to-r-and-posit-cloud---week-2a" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>You should aim to complete this chapter in Week 2, Semester 1.</p>
<div id="what-is-r-and-posit-cloud" class="section level2 hasAnchor" number="2.1">
<h2 class="hasAnchor"><span class="header-section-number">2.1</span> What is R and posit Cloud?<a href="#what-is-r-and-posit-cloud" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>So hopefully you have read through Chapter 1, where R and the posit Cloud interface were introduced in this module. But what really is R and what is the difference between R and posit Cloud?</p>
<p>Essentially R is a programming language that is commonly used in statistical computing, data handling, data visualisation and data analysis. Posit Cloud is a cloud based interface for a piece of software called RStudio (we wont be using the non cloud based RStudio in this sub-module, so we wont explore this software further here). Posit Cloud uses the R programming language but has a nice user friendly interface and is a great tool for learning how to conduct analysis in R.</p>
<p>We are using the posit cloud rather then RStudio because it means that no one has to worry about installing extra software on their own computers and everyone will be working with the same software versions. You will all need to create your own free accounts on posit Cloud, but first have a look at the short video below introducing you to the interface.</p>
<div style="position:relative;height:0;padding-bottom:40%">
<iframe width="560" height="315" src="https://www.youtube.com/embed/3l02OgYYb0I" title="Intro to posit Cloud" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
</iframe>
</div>
<p>
<p>Now that you’ve watched the video, create your own Free posit Cloud account <a href="https://login.rstudio.cloud/login?redirect=%2Foauth%2Fauthorize%3Fredirect_uri%3Dhttps%253A%252F%252Frstudio.cloud%252Flogin%26client_id%3Drstudio-cloud%26response_type%3Dcode%26show_auth%3D0%26show_login%3D1">here</a>.</p>
</div>
<div id="a-warning-on-managing-your-work-flow" class="section level2 hasAnchor" number="2.2">
<h2 class="hasAnchor"><span class="header-section-number">2.2</span> A warning on managing your work flow<a href="#a-warning-on-managing-your-work-flow" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Your free posit Cloud account comes with some restrictions which you should be aware of. These are;</p>
<ul>
<li>You may have up to 50 projects in total</li>
<li>You are limited to 25 project hours per month</li>
<li>You have up to 1GB of RAM and 1 CPU per project</li>
</ul>
<p>For the purposes of our work here these restrictions should not be a problem. But I strongly suggest you don’t leave your coursework to the last minute. If you try and pack in a whole semester or years worth or material into a single month, you will brush up against those 25 project hours. There are some options of last resort should this happen, you could install RStudio on your desktop for example but things will get complicated. Lets avoid that where we can!There are some ways you can increase the efficiency of your workspace which are documented below in section 2.4.</p>
</div>
<div id="creating-your-first-posit-cloud-project" class="section level2 hasAnchor" number="2.3">
<h2 class="hasAnchor"><span class="header-section-number">2.3</span> Creating your first posit Cloud Project<a href="#creating-your-first-posit-cloud-project" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Once you have created an posit Cloud account you should be presented with this window</p>
<div class="figure">
<img src="figures/screenshot_rstudiocloud_1.jpg" alt="Screen for when you first login to posit Cloud" width="90%" />
<p class="caption">
(#fig:unnamed-chunk-1)Your workspace in posit Cloud
</p>
</div>
<p>Under <strong>Spaces</strong> go to <strong>Your Workspace</strong> and under <strong>Projects</strong> create a <strong>New Project &gt; New R studio Project</strong>.</p>
<div class="figure">
<img src="figures/screenshot_rstudiocloud_2.png" alt="Screen for creating a new project" width="90%" />
<p class="caption">
(#fig:unnamed-chunk-2)Creating a new project in posit Cloud
</p>
</div>
<p>Lets name this project <code>testing_R</code>, notice that I have no spaces in my project name. Instead of a space I have used an underscore, there are a number of good habits you should try and adopt when project or file naming and not including spaces is one of them, we will go over this in more depth in lectures and later chapters. You can rename your project by clicking on <strong>Untitled Project</strong> at the top of the window, and typing in your new project name.</p>
<div class="figure">
<img src="figures/screenshot_rstudiocloud_3.jpg" alt="Screen for renaming your project" width="90%" />
<p class="caption">
(#fig:unnamed-chunk-3)Renaming your project
</p>
</div>
<p>You will see that your new project has three panels with tabs showing the Console, Environment and Files for your project.</p>
<div class="figure">
<img src="figures/screenshot_rstudiocloud_4.jpg" alt="Screen for your r studio workspace" width="90%" />
<p class="caption">
(#fig:unnamed-chunk-4)The three basic panels of an R Studio workspace
</p>
</div>
</div>
<div id="maximising-the-efficiency-of-your-workspace" class="section level2 hasAnchor" number="2.4">
<h2 class="hasAnchor"><span class="header-section-number">2.4</span> Maximising the efficiency of your workspace<a href="#maximising-the-efficiency-of-your-workspace" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We know we have limited project hours on our free posit Cloud accounts to reduce your resource usage. In your project, go to settings (the cogwheel sign), Resources, and drag all of the sliders down to the minimum setting, apply the changes, this will mean the project has to restart. This will mean that running the project for 1 hour will consume 0.5 project hours. We wont be doing anything particularly computationally intensive so there is no need for RAM of CPUs to be higher.</p>
<div class="figure">
<img src="figures/screenshot_rstudiocloud_5.png" alt="Making your project more efficient" width="90%" />
<p class="caption">
(#fig:unnamed-chunk-5)How to make your project more efficient
</p>
</div>
</div>
<div id="entering-commands-directly-into-the-consol" class="section level2 hasAnchor" number="2.5">
<h2 class="hasAnchor"><span class="header-section-number">2.5</span> Entering commands directly into the consol<a href="#entering-commands-directly-into-the-consol" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Lets start by playing with some commands in the console. Type or copy and paste the simple calculation (or command) shown below, into the console. Your text should appear next to the <code>&gt;</code> symbol. Press <code>Enter</code> on your keyboard, this will instruct R to run the command.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="dv">5</span> <span class="sc">+</span> <span class="dv">9</span></span></code></pre></div>
<p>The two final lines of your output should look like this:</p>
<pre><code>&gt; 5 + 9
[1] 14</code></pre>
<p>So… your initial command <code>5 + 9</code> is shown after the <code>&gt;</code>, symbol and the resulting output from R is shown after <code>[1]</code>. Don’t worry too much about the syntax of <code>&gt;</code> and <code>[1]</code> here. You can think of <code>&gt;</code> as meaning that R is ready to receive a command and <code>[1]</code> as R telling you that the answer to the first part of your question is here (in this case <code>14</code>).</p>
<p>Try out some other commands… What happens when you input the following?</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="dv">362</span> <span class="sc">*</span> <span class="dv">12</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="dv">55</span> <span class="sc">/</span> <span class="dv">5</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>(<span class="dv">40</span> <span class="sc">/</span> <span class="dv">990</span>) <span class="sc">*</span> <span class="dv">100</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="dv">4</span><span class="sc">^</span><span class="dv">2</span></span></code></pre></div>
<p>See if you can work out what 30% of 735 is using the R console</p>
</div>
<div id="sending-commands-down-to-the-console-from-an-r-script" class="section level2 hasAnchor" number="2.6">
<h2 class="hasAnchor"><span class="header-section-number">2.6</span> Sending commands down to the console from an R script<a href="#sending-commands-down-to-the-console-from-an-r-script" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>I have already mentioned the <strong>reproducibility</strong> factor as an advantage of using posit Cloud. This is because you can record and run all of your commands from an R script within posit Cloud. This means that you have a written record of your analysis workflow, what you did to your data at which stage, and you can do all of this without altering the original data files! This is super important because it means that if you revisit your work in a few weeks/months/years you can see exactly what you did AND if someone else needs to rerun any of your analysis or use your workflow on some other data, they can!</p>
<p>So lets go about setting up your new script. You currently have three panels in posit Cloud. If you go to <strong>file &gt; New File &gt; R Script</strong> a new panel will open.</p>
<div class="figure">
<img src="figures/screenshot_rstudiocloud_6.png" alt="creating a new r script" width="90%" />
<p class="caption">
(#fig:unnamed-chunk-11)How to create a new R script
</p>
</div>
<div class="figure">
<img src="figures/screenshot_rstudiocloud_7.jpg" alt="Screen for your r studio workspace" width="90%" />
<p class="caption">
(#fig:unnamed-chunk-12)The four panels of an R Studio workspace
</p>
</div>
<p>This new panel is essentially a text file where you can write your commands into a script and then send them down to the console when you want to run them. Lets have a play. Copy the below into your new R script.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="dv">362</span> <span class="sc">*</span> <span class="dv">12</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="dv">55</span> <span class="sc">/</span> <span class="dv">5</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>(<span class="dv">40</span> <span class="sc">/</span> <span class="dv">990</span>) <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="dv">4</span><span class="sc">^</span><span class="dv">2</span></span></code></pre></div>
<p>These are all commands you have run before but now if you save your script you will have a text based document with your progress saved. It doesn’t make a huge amount of sense to do this now because these are just a un-associated set of calculations and we are just playing with the interface, but you could if you wanted to.</p>
<p>Ok, so now we are ready to execute our commands. You can do this with each calculation individually, line by line, or you can run the whole script.</p>
<p>To run your script line by line, place your cursor on the line you wish to run and you can either;</p>
<ul>
<li>Click on the <strong>Run</strong> button on the top right of the script panel</li>
<li>Press <strong>ctrl + Enter</strong> (or <strong>Command + Enter</strong> on mac)</li>
</ul>
<p>Or if you wish to run the entire script you can either;</p>
<ul>
<li>Manually highlight the script and click the <strong>Run</strong> button or press <strong>ctrl + Enter</strong> (or <strong>Command + Enter</strong> on mac)</li>
<li>Press <strong>ctrl + A</strong> (or <strong>Command + A</strong> on mac) and then click the run button or press <strong>ctrl + Enter</strong> (or <strong>Command + Enter</strong> on mac)</li>
</ul>
</div>
<div id="a-note-on-comments" class="section level2 hasAnchor" number="2.7">
<h2 class="hasAnchor"><span class="header-section-number">2.7</span> A note on comments<a href="#a-note-on-comments" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>When you are writing a script you can leave notes for yourself. This can be extremely useful if you need reminding on what a particular piece of script is doing. This is a good learning tool while you are still learning how to use R, but its also a good habit to get into for later. As you progress with R you will inevitably start constructing quite intricate scripts, leaving good notes on what each part of the script is doing is very useful for you and for anyone else who may later come to use your script. You can leave notes or comments in scripts if you precede them with <code>#</code>. If R sees <code>#</code> it will ignore all text on that line that comes after it. For example, copy and paste the following into your script and try running it line by line.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Four squared can be calculated with the following command</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="dv">4</span><span class="sc">^</span><span class="dv">2</span></span></code></pre></div>
<p>In this case I have reminded myself how to calculate 4 squared, everything after the <code>#</code> will be ignored by R when you try to run the line.</p>
<p>Well Done! You have started writing and running simple code. But R is capable of doing so much more than simple calculations. Next week in Chapter 3 we will explore how we can create objects, and use packages and functions.</p>
<!--chapter:end:01-intro_to_R.Rmd-->
</div>
</div>
<div id="objects-functions-and-packages---week-3a" class="section level1 hasAnchor" number="3">
<h1 class="hasAnchor"><span class="header-section-number">3</span> Objects, Functions and Packages - Week 3A<a href="#objects-functions-and-packages---week-3a" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>You should aim to complete this chapter in Week 3, Semester 1</p>
<div id="creating-objects" class="section level2 hasAnchor" number="3.1">
<h2 class="hasAnchor"><span class="header-section-number">3.1</span> Creating objects<a href="#creating-objects" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As we saw in Chapter 2, R can be used to perform calculations. However, we can use R to perform tasks that are much more complex. In order to do this we are going to have to learn about objects.</p>
<p>Log in to your posit Cloud account and go into your <code>testing_R</code> project. Create a new R script and save it under the name <code>objects</code>. You should see it appear as <code>object.R</code> under the <strong>Files</strong> tab in the bottom right panel. Run the following command using your script.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>A <span class="ot">&lt;-</span> <span class="dv">50</span></span></code></pre></div>
<p>What you have essentially done here, is told R to create an object called <code>A</code> and to store the number <code>50</code> inside it using the syntax <code>&lt;-</code>. After running the command you should see your object <code>A</code> and <code>50</code> pop up under <strong>Environment</strong> and <strong>Values</strong> in the top right hand panel, so you can see that <code>50</code> has been stored in <code>A</code>.</p>
<p>Lets create some more objects</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>B <span class="ot">&lt;-</span> <span class="dv">6</span></span></code></pre></div>
<p>Now see what happens if you run the following command</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>A <span class="sc">*</span> b</span></code></pre></div>
<p>Oh dear… we have an error…</p>
<pre><code>&gt; A * b
Error: object &#39;b&#39; not found</code></pre>
<p>This is because R has absolutely no flexibility with typos. It is looking for an object called <code>b</code> when there is no object called <code>b</code> in your environment. There is an object called <code>B</code>, but to R, <code>B</code> and <code>b</code> are completely different things. Try running this instead.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>A <span class="sc">*</span> B</span></code></pre></div>
<p>Hopefully you should now have an output that looks like this</p>
<pre><code>&gt; A * B
[1] 300</code></pre>
<p>This is a very simple example. But lets try to demonstrate why saving values within objects may be useful. Lets say you are interested in looking at the number of students who get freshers flu in Week 1 of teaching at UEA. You have a class size of 200 students and 15 of them report that they have contracted freshers flu. Use the following script to calculate the percentage of students with freshers flu.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create two objects, one for your total class and one for those that have flu</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Notice that neither of my object names contain spaces</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co"># If you need a space always use an underscore or full stop</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>total_class <span class="ot">&lt;-</span> <span class="dv">200</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>with_flu <span class="ot">&lt;-</span> <span class="dv">15</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the percentage of students with flu in week 1</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>percentage_with_flu <span class="ot">&lt;-</span> (with_flu<span class="sc">/</span>total_class)<span class="sc">*</span><span class="dv">100</span></span></code></pre></div>
<p>If you look in your Environment in the top right panel you should see the percentage of students with flu calculated and stored under <code>percentage_with_flu</code>. You can use the following command to see it printed in the consol.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(percentage_with_flu)</span></code></pre></div>
<p>Some students have been late in reporting their symptoms. A week later you hear that <code>7</code> more students also had freshers flu in Week 1. If you had been typing into the console and not using objects you would have to type this script out all over again. But you don’t need to. You just need to change your entry for <code>with_flu</code> from <code>15</code> to <code>22</code>. Do that now and re-run the script. By editing the <code>15</code>, R will overwrite the object <code>with_flu</code> to represent the new value of <code>22</code>. If you run the last line of code then the percentage of students with flu will also update.</p>
<p>This may seem like a very simple example, and it is, but we are still building up your knowledge of R so you will have to take my word for it that objects will be very helpful to you as we progress :). You will also see, that objects can contain a range of different types of data. We will cover some of these in lectures, but for now lets leave objects and have a think about functions and packages.</p>
</div>
<div id="what-are-functions-and-packages" class="section level2 hasAnchor" number="3.2">
<h2 class="hasAnchor"><span class="header-section-number">3.2</span> What are functions and packages?<a href="#what-are-functions-and-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If you have the coding skills it is possible to do pretty much whatever you like with your data in R. However, why reinvent the wheel trying to write your own complex scripts, when a lot of very clever coders have already written lots of functional bits of code, known as <strong>functions</strong>, for general use. Functions can be thought of as a piece of code that is designed to perform a set task. R comes with lots of functions already built in, but there are also lots of additional functions that are stored in <strong>packages</strong>.</p>
<p>Packages are containers that can hold sets of functions or data and as the course progresses you will use a range of packages that contain useful functions and data. For example the data visualisation package (which you will become very familiar with later on) <code>ggplot2</code> contains ranges of functions which allow you to define how your plot or graph will look, for example <code>geom_bar</code> is a function within the <code>ggplot2</code> package that contains the instructions required to build a bar chart.</p>
<div id="using-functions" class="section level3 hasAnchor" number="3.2.1">
<h3 class="hasAnchor"><span class="header-section-number">3.2.1</span> Using functions<a href="#using-functions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We actually have already had some exposure to functions. You used the print function earlier when you asked R to print out the value contained within the <code>percentage_with_flu</code> object. But lets have a go at using another function.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># First of all we need some data</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Notice the syntax, by using c() I have told R to prepare for a list of values</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Each value is separated by a comma</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">6</span>,<span class="dv">8</span>,<span class="dv">10</span>,<span class="dv">12</span>,<span class="dv">14</span>,<span class="dv">16</span>,<span class="dv">18</span>,<span class="dv">20</span>)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Now I want to know what the total value of data is if I add all the values stored within it together</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>total <span class="ot">&lt;-</span> <span class="fu">sum</span>(data)</span></code></pre></div>
<p>In the above piece of code we have created some data, as a list of values, and stored them under the object name data. We have then used the function <code>sum()</code>. We have essentially told R that we wish it to perform the function <code>sum()</code> on everything contained within the object <code>data</code> and store it in another object called <code>total</code>. You can see the value within <code>total</code> in your environment, can you work out how to get R to print the value of <code>total</code> out?</p>
<p>If you ever need help working out how to use a function you can use another function <code>help()</code>. Try inputing the code below into your console.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">help</span>(sum)</span></code></pre></div>
<p>This should bring up a help file in your bottom right hand panel that describes and explains the use of the function <code>sum()</code>.</p>
</div>
</div>
<div id="installing-packages" class="section level2 hasAnchor" number="3.3">
<h2 class="hasAnchor"><span class="header-section-number">3.3</span> Installing packages<a href="#installing-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>sum</code> function is part of the base R package that comes with any R installation. However there are lots of other useful functions that are held in additional packages. In order to use these functions, you must have the necessary packages installed in your work space. I already mentioned that we would be using a package called <code>ggplot2</code>, lets try installing and loading it in your work space.</p>
<p>We only need to install a package once so copy this into your console</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggplot2&quot;</span>)</span></code></pre></div>
<p>Now we can try to make a simple plot using a function from within <code>ggplot2</code> and an example data set from within the base R package called <code>iris</code>.</p>
<p>Copy the following command into your script and run it</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># First load ggplot and the base R datasets in your workspace</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;datasets&quot;</span>)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Then call the ggplot function and tell it that the data set you wish to use is calles iris</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co"># We then need to tell ggplot how to map our variables</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="co"># The aes function is an aesthetic tool that allows us to define variables for the x and y axes</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="co"># We then use the geom_point function to instruct R to built a scatter plot</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a><span class="co"># A second use of the aes function tells R to color points by species</span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> iris,<span class="fu">aes</span>(<span class="at">x =</span> Sepal.Length, <span class="at">y =</span> Sepal.Width)) <span class="sc">+</span> </span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour=</span>Species)) </span></code></pre></div>
<p>Those sharped eyed among you may have noticed that the short script defining our plot goes over two lines, has a <code>+</code> and an indentation on the second line. The <code>+</code> and indentation is essentially allowing you to pipe one line of code into the next, adding more directions in how you would like your graph to be built and linking together multiple functions. Don’t worry if this is a lot to take in at this stage, as we progress with <code>ggplot2</code> we will revisit this in more detail.</p>
</div>
<div id="what-do-i-do-if-i-get-an-error-message" class="section level2 hasAnchor" number="3.4">
<h2 class="hasAnchor"><span class="header-section-number">3.4</span> What do I do if I get an error message?<a href="#what-do-i-do-if-i-get-an-error-message" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Error messages are almost guaranteed when working regularly with R, they are super common, dont panic when you eventually run into one. In fact you may remember that we already ran into one earlier in this chapter, that particular error was caused by a typo. Generally the most common causes of error messages are;</p>
<ul>
<li>A typo (you can reduce risk of this by consistently labeling your files and objects)</li>
<li>A mistake in syntax (missing or un-closed; <code>()</code>, <code>""</code>, or missing <code>,</code>)</li>
<li>Missing dependencies (maybe you didnt install or load a package correctly, or maybe if your running a longer script, you are trying to call an object that you havent created yet)</li>
</ul>
<p>This is by no means an exhaustive list but when I run into errors its normally because of one of the above. The trick is don’t panic and read the error message carefully, it often tells you what is wrong. Then look over your script slowly to troubleshoot. And if you are genuinely stuck and the error doesn’t make sense, run it through Google, I can almost guarantee you wont have been the first person to have that error message.</p>
<p>Well done everyone. I hope you have enjoyed this weeks session. Next week is our first workshop and we will start to use some of these new found skills in tandem and start to look at using them on some new data.</p>
</div>
<div id="references" class="section level2 hasAnchor" number="3.5">
<h2 class="hasAnchor"><span class="header-section-number">3.5</span> References<a href="#references" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Wickham, Hadley, Winston Chang, Lionel Henry, Thomas Lin Pedersen, Kohske Takahashi, Claus Wilke, Kara Woo, Hiroaki Yutani, and Dewey Dunnington. 2021. Ggplot2: Create Elegant Data Visualisations Using the Grammar of Graphics. <a href="https://CRAN.R-project.org/package=ggplot2" class="uri">https://CRAN.R-project.org/package=ggplot2</a>.</p>
<!--chapter:end:02-packages_and_functions.Rmd-->
</div>
</div>
<div id="directories-and-data-frames---week-4a---workshop-1" class="section level1 hasAnchor" number="4">
<h1 class="hasAnchor"><span class="header-section-number">4</span> Directories and data frames - Week 4A - Workshop 1<a href="#directories-and-data-frames---week-4a---workshop-1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This week we don’t have any lectures but we do have an in person workshop. The section to complete this week may be a little more chunky then unusual, complete what you can during the workshop and then finish it off in your own time. Do make use of the demonstrators if any of the exercises so far haven’t made sense.</p>
<div id="data-entry" class="section level2 hasAnchor" number="4.1">
<h2 class="hasAnchor"><span class="header-section-number">4.1</span> Data Entry<a href="#data-entry" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Lets have a look at some actual data now, Figure 4.1 shows some lab notes taken during the data collection phase of a small research project to investigate parasite load in 2 small populations of the flour beetle <em>Tribolium castaneum</em>. Each beetle has an ID, a measure of wing casing in mm (which is a good proxy for overall size) and a tally for the number of tapeworm cysticercoids dissected out. In terms of data handling and analysis, we need this data to be stored digitally and we need to have it ordered sensibly in a spreadsheet. We covered what to look for in a tidy, well ordered spreadsheet in our week 3 lecture. Have a go at entering the data from Figure 4.1 into an Excel spreadsheet. When you’re done go to <strong>File &gt; Save As…</strong>, name your spreadsheet something sensible and make sure you select a sensible place to save it to, where it says <strong>File Format:</strong> select <strong>Comma-separated Values (.csv)</strong>.</p>
<div class="figure">
<img src="figures/parasite_burden.jpg" alt="Tribolium parasite data raw" width="90%" />
<p class="caption">
(#fig:unnamed-chunk-25)Lab notes for wing case size (mm) and parasite burden of 20 <em>Tribolium castaneum</em>
</p>
</div>
<div id="a-note-on-naming" class="section level3 hasAnchor" number="4.1.1">
<h3 class="hasAnchor"><span class="header-section-number">4.1.1</span> A note on naming<a href="#a-note-on-naming" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>While we are on the subject of naming, I am going to take a moment to go over a few good rules to follow. You have just created names for some variables and for a file. You will also frequently have to name folders/directories and objects while working in R, so naming is something that you are repeatedly going to have to do. It therefore makes sense for you to learn how to do it well.</p>
<ul>
<li><p>Keep names meaningful - Over the course of your degree you will create a lot of files and folders/directories. You will also frequently be naming data variables and objects. You need to be able to easily recognise what is what so informative names are better then arbitrary names. So <code>girraffe</code>, <code>elephant</code>, <code>hippo</code> are better names then <code>1</code>, <code>2</code> and <code>3</code>.</p></li>
<li><p>Make versions obvious - There will be occasions when you want to save new information in a file or object but keep the original untouched. You may be experimenting or you may want the option to revisit to your original work. In these instances it makes sense to save your edits to a new version of the original file or object. If you do this and especially if you are sharing it with others make it clear. I tend to add my initials and a version identifier (e.g. v1, v2) as a suffix to the file name.</p></li>
<li><p>Keep names short - Typing takes time, its boring, and the more you type the greater the risk of a typo. So keep names short. So <code>trunk_length</code> is better than <code>elephant_trunk_length_cm</code>, there is important information in the second name but you can store this as a comment or a note.</p></li>
<li><p>Avoid spaces - Lots of software packages hate spaces, including R. If you can get out of the habbit of using spaces and instead use <code>_</code> your future self will thank you.</p></li>
<li><p>Abide by the rules for R - R does have some rules of its own for naming. Names must <strong>not</strong> start with a number, but numbers may be used provided the first character is a letter. You can also use <code>.</code> or <code>_</code>, but avoid spaces at all costs. You may use upper or lower case letters, but remember R is case sensitive so will see <code>Elephant</code> and <code>elephant</code> as different. I try to keep to a lower case and use <code>_</code> instead of spaces, it keeps everything uniform and saves me time (this style is known as snake_case).</p></li>
</ul>
<p>Now go back to your spreadsheet, do your variables and file names follow a sensible naming convention?</p>
<p><strong>Make sure you have a demonstrator check your work before moving onto the next step.</strong></p>
</div>
</div>
<div id="setting-up-your-r-workspace" class="section level2 hasAnchor" number="4.2">
<h2 class="hasAnchor"><span class="header-section-number">4.2</span> Setting up your R workspace<a href="#setting-up-your-r-workspace" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This is a new piece of analysis, we will be carrying forward principals from the previous chapters but scripts and data will be new. So its a good idea to start with a fresh clean workspace. So lets set up a new project in posit Cloud, follow the steps from Chapter 2 if you cant remember how to do this and name this project <code>tribolium_parasites</code>. Its always good to keep a well organised tidy workspace, generally I will start a new project in posit Cloud if I am working on a new series of questions with a new data set or sets.</p>
<p>For this project you need to make sure two packages are installed. Copy the following directly into your console.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;tidyverse&quot;</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;janitor&quot;</span>)</span></code></pre></div>
<p>We will be moving towards doing some more comprehensive data handling so setting up your workspace is going to become more and more important. If you can get into the habit of setting up your workspace properly at the start of each project you will thank yourself later. A well developed project will contain several files including;</p>
<ul>
<li>Input data</li>
<li>Scripts</li>
<li>Outputs</li>
</ul>
<p>Under <strong>Files</strong> in the bottom right panel create a <strong>New Folder</strong>, for each of the following;</p>
<ul>
<li>data</li>
<li>scripts</li>
<li>figures</li>
</ul>
<p><strong>Make sure you copy the names over exactly, or some of the commands that we will be using later wont work</strong></p>
<p>A tidy workspace will help you find the things you need when you need them and will reduce the risk of loosing things.</p>
<div id="a-note-on-directories-and-their-structure" class="section level3 hasAnchor" number="4.2.1">
<h3 class="hasAnchor"><span class="header-section-number">4.2.1</span> A note on directories and their structure<a href="#a-note-on-directories-and-their-structure" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Working with text based interfaces like R, where you are writing and running commands, means that we need to have some understanding of the underlying computer architecture. You need to have some appreciation for how files are organised within folders/directories and why its important to know which directory you are working from with R.</p>
<p>Under <strong>Files</strong> in the bottom right panel of your screen you will see a file with the ending .Rproj. This is an R project file which tells R where you are working in the server. It means that R will automatically treat your project location as the <strong>working directory</strong>. If you wish to access files in any of the folders you have just created you will need to tell R the <strong>path</strong> that it needs to follow in order to access these files. There are two types of path;</p>
<ul>
<li><strong>absolute</strong> paths, which refer to the same location in a file system relative to the root directory (don’t worry if that doesn’t make too much sense at this stage).</li>
<li><strong>relative</strong> paths, which refer to a location in the file system relative to the directory you are currently working in.</li>
</ul>
<p>Here we will only be dealing with <strong>relative</strong> paths because you will always be working from within your R project directory. So for example if you wish to access a file in your <code>data</code> folder in the R project you would need to provide the path <code>("data/any_sub_folders_you_may_have_made/the_file_you_wish_to_access")</code>. We will do this in the next step so don’t worry if you don’t quite get it yet.</p>
</div>
</div>
<div id="setting-up-your-script" class="section level2 hasAnchor" number="4.3">
<h2 class="hasAnchor"><span class="header-section-number">4.3</span> Setting up your script<a href="#setting-up-your-script" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Right OK, enough theory, lets start setting up your script. Create a new R script and call it <code>tribolium_parasites</code>. Its always good to add a few lines of description at the start of your script, this tells your future self and anyone else what the script is doing. Copy over the following.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># An analysis of parasite abundance between male and female tribolium beetles from two thermal treatments (30 and 35 degrees centigrade)</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Your name</span></span></code></pre></div>
<p>Then we need to load any packages that we might want to use, copy over the following;</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse) <span class="co"># Loads the tidyverse package</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(janitor) <span class="co"># Loads the janitor package</span></span></code></pre></div>
</div>
<div id="importing-data" class="section level2 hasAnchor" number="4.4">
<h2 class="hasAnchor"><span class="header-section-number">4.4</span> Importing data<a href="#importing-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Lets import your data set. Go into your <strong>data</strong> folder, click <strong>Upload</strong> and select your newly made <code>.csv</code> file. Once its uploaded you should be able to see it under <code>data/</code> in the bottom right panel.</p>
<p>Now we need to tell R to read your <code>.csv</code> file.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the data from the .csv file in the data folder into my project</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>tribolium <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/your_file_name.csv&quot;</span>) <span class="co">#</span></span></code></pre></div>
<p>R should respond with this;</p>
<pre><code>&gt; tribolium &lt;- read_csv(&quot;data/tribolium.csv&quot;)
Rows: 20 Columns: 3                                                              
── Column specification ───────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
dbl (3): beetle_id, wing_case_size, parasite_burden</code></pre>
<p>Here R has told us that there are 20 rows and 3 columns in our data set, it has indicated that it has used comas as a column deliminator and that all the data are dbl (short for double), which means each column contains numbers. You should also be able to see <code>tribolium</code> in your <strong>Environment</strong> under <strong>Data</strong>.</p>
<p>If you have errors, make sure your file name and path are input correctly.</p>
<p>Lets have a look at our data to make sure it looks as we expect it to, try this;</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tribolium)</span></code></pre></div>
<p>We can see for ourselves that R has correctly read our data.</p>
</div>
<div id="creating-tibbles" class="section level2 hasAnchor" number="4.5">
<h2 class="hasAnchor"><span class="header-section-number">4.5</span> Creating tibbles<a href="#creating-tibbles" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>So we have successfully imported data into R from a <code>.csv</code> file and stored it as a data frame which is a type of two dimensional array. Tibbles are a type of data frame as well, these can be useful if you want to type your data straight into R, or if you have more information to add to a pre-existing data frame. For example, we also have some additional data on the sex and treatment conditions of the Tribolium beetles. So lets create a tibble, add the following to your script;</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># First of all lets create some variables</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>beetle_id <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>) <span class="co"># Notice that this notation will create of a list of numbers 1 to 20</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>sex <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>)</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>treatment <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>)</span></code></pre></div>
<p>Now we neet to turn our new variables into a data frame</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Turning variables into a tibble</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>tribolium_extras <span class="ot">&lt;-</span> <span class="fu">tibble</span>(beetle_id,sex,treatment)</span></code></pre></div>
<p>Try using <code>head()</code> to check your new `tribolium_extras’ data frame.</p>
<p>It may be advantageous to have all the data from both of our data frames together, we can use another function to do this. However first I would like everyone in the class to have the same variable names, this will help us later on but also sometimes its useful to change headings in R. Add the following to your script and run it, make sure you edit it to match your variable names.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Renaming some variables in your original data set</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>tribolium <span class="ot">&lt;-</span> tribolium <span class="sc">%&gt;%</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">beetle_id =</span> your_variable_name_here)</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check how your data frame now looks</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tribolium) </span></code></pre></div>
<p>Side note - you may have noticed the %&gt;% expression, this is known as the forward pipe operator and allows you to chain together commands (rather like the <code>+</code> we covered in Chapter 2 when using <code>ggplot</code>). This is a very simple use of it but it is piping your data frame forward to the rename function.</p>
<p>Hopefully the column containing your beetle IDs is now labeled <code>beetle_id</code>. Use the same set of commands to rename your variables for beetle wing case size and parasite load. Call the variables <code>wing_case_size</code> and <code>parasite_burden</code> instead.</p>
<p>Now we can merge our two data frames.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge data frames</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>tribolium_combo <span class="ot">&lt;-</span> <span class="fu">merge</span>(tribolium, tribolium_extras, <span class="at">by =</span> <span class="st">&quot;beetle_id&quot;</span>)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="co"># This merge function combines both of your data frames but matches rows by beetle_id</span></span></code></pre></div>
<p>Take a look at your new data frame now with the <code>head()</code> function, and lets try one more command to do one last check on your data. Try this;</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(tribolium_combo)</span></code></pre></div>
<p>This command will show you something like this;</p>
<pre><code>&gt;glimpse(tribolium_combo)
Rows: 20
Columns: 5
$ beetle_id       &lt;dbl&gt; 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17…
$ wing_case_size  &lt;dbl&gt; 2.4, 2.2, 2.5, 2.3, 2.3, 2.6, 2.6, 2.3, 2.4, 2.4, 2.3, 2.…
$ parasite_burden &lt;dbl&gt; 15, 15, 19, 13, 12, 19, 22, 10, 14, 16, 11, 8, 15, 9, 14,…
$ sex             &lt;chr&gt; &quot;F&quot;, &quot;F&quot;, &quot;F&quot;, &quot;F&quot;, &quot;F&quot;, &quot;M&quot;, &quot;M&quot;, &quot;M&quot;, &quot;M&quot;, &quot;M&quot;, &quot;F&quot;, &quot;F…
$ treatment       &lt;dbl&gt; 35, 35, 35, 35, 35, 35, 35, 35, 35, 35, 30, 30, 30, 30, 3…</code></pre>
<p>Once again we can see that four of our variables have been interpreted by R as being <code>&lt;dbl&gt;</code> vectors, R has identified them as being numeric. One of our variables has been identified as being a <code>&lt;chr&gt;</code> (short for character) vector, so R knows that this variable contains letters or words.</p>
<p>You may now wish to export your new combined data frame as a <code>.csv</code> file. You can do this by running the following command;</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save your combined data frame as a .csv file (which you could download) to your data folder</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(tribolium_combo,<span class="st">&quot;data/tribolium_combo.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>Well done everyone, you have successfully entered your data and uploaded it to posit Cloud as well as making some small adjustments to the data frame and doing a first pass quality check! There are more checks we could have run, and we will revisit this later in the course when we look at some larger data sets. But we will leave this here for now, next week is reading week and the week after we will start looking at checking our data dimensions.</p>
</div>
<div id="before-you-leave" class="section level2 hasAnchor" number="4.6">
<h2 class="hasAnchor"><span class="header-section-number">4.6</span> Before you leave!<a href="#before-you-leave" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Log out of posit Cloud and make sure you save your script!</p>
</div>
<div id="references-1" class="section level2 hasAnchor" number="4.7">
<h2 class="hasAnchor"><span class="header-section-number">4.7</span> References<a href="#references-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Firke, Sam. 2021. Janitor: Simple Tools for Examining and Cleaning Dirty Data. <a href="https://github.com/sfirke/janitor" class="uri">https://github.com/sfirke/janitor</a>. <br />
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. “Welcome to the tidyverse.” Journal of Open Source Software 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686" class="uri">https://doi.org/10.21105/joss.01686</a>. <br />
Wickham, Hadley, Romain François, Lionel Henry, and Kirill Müller. 2021. Dplyr: A Grammar of Data Manipulation. <a href="https://CRAN.R-project.org/package=dplyr" class="uri">https://CRAN.R-project.org/package=dplyr</a>. <br />
Wickham, Hadley, Jim Hester, and Jennifer Bryan. 2021. Readr: Read Rectangular Text Data. <a href="https://CRAN.R-project.org/package=readr" class="uri">https://CRAN.R-project.org/package=readr</a>.</p>
<!--chapter:end:03-directories_and_dataframes.Rmd-->
</div>
</div>
<div id="reading-week---week-5a" class="section level1 hasAnchor" number="5">
<h1 class="hasAnchor"><span class="header-section-number">5</span> Reading week - Week 5A<a href="#reading-week---week-5a" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>There will be no new material released this week, we recommend that you use this time to catch up on any former sections of the workbook that you may not have completed yet.</p>
<!--chapter:end:04-reading_week.Rmd-->
</div>
<div id="exploring-the-dimensions-of-your-data-frame---week-6a" class="section level1 hasAnchor" number="6">
<h1 class="hasAnchor"><span class="header-section-number">6</span> Exploring the dimensions of your data frame - Week 6A<a href="#exploring-the-dimensions-of-your-data-frame---week-6a" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>You should aim to complete this chapter in Week 6, Semester 1</p>
<div id="top-to-tail---some-useful-functions-to-explore" class="section level2 hasAnchor" number="6.1">
<h2 class="hasAnchor"><span class="header-section-number">6.1</span> Top to tail - some useful functions to explore<a href="#top-to-tail---some-useful-functions-to-explore" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Log into posit cloud and open up your <code>tribolium_parasites</code> project.</p>
<p>Lets start by reminding ourselves of what these data look like. Try using the following functions on your<code>tribolium_combo</code> data frame, copy them into your script and comment <code>#</code> next to each when you work out what it does.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Some functions to play with</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>()</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ncol</span>()</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>()</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>()</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="fu">view</span>()</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>()</span></code></pre></div>
<p>Hopefully you can see that these functions help give you some good insights into the dimensions of your data set, especially when used alongside functions that you are already be familiar with, like <code>head()</code> and <code>glimpse()</code>.</p>
</div>
<div id="whats-your-sample-size" class="section level2 hasAnchor" number="6.2">
<h2 class="hasAnchor"><span class="header-section-number">6.2</span> Whats your sample size?<a href="#whats-your-sample-size" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>With this data set its not hard to work out what the sample size is. But data sets can become huge and its not always easy or effective to scroll through them. Sometimes you really just need to know how many observations are in your data set, thankfully each of our beetles has a unique identifyer so we can count how many unique entries there are in the <code>beetle_id</code> variable. Try the following;</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the n_distinct function to count unique/distinct entries in the beetle_id column</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">n_distinct</span>(tribolium_combo<span class="sc">$</span>beetle_id)</span></code></pre></div>
<p>Notice the use of the <code>$</code> above. This is effectively directing R to a specific variable/column within a data frame.</p>
<p>You may have noticed that we have two categorical variables in this data set, <code>sex</code> and <code>treatment</code>. It would be really useful to know what your sample size is for each of these variables. Try the following;</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the tabyl function from the janitor package, comment on what it does. </span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tabyl</span>(tribolium_combo, sex , treatment)</span></code></pre></div>
<p>So hopefully you can see from the <code>summarise</code> and <code>tabyl</code> functions that we have a total sample size of 20 beetles and that 5 of each sex were exposed to each treatment. Next week we will start to explore how we can visualise some of these data.</p>
</div>
<div id="before-you-leave-1" class="section level2 hasAnchor" number="6.3">
<h2 class="hasAnchor"><span class="header-section-number">6.3</span> Before you leave!<a href="#before-you-leave-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Log out of posit Cloud and make sure you save your script!</p>
</div>
<div id="references-2" class="section level2 hasAnchor" number="6.4">
<h2 class="hasAnchor"><span class="header-section-number">6.4</span> References<a href="#references-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Firke, Sam. 2021. Janitor: Simple Tools for Examining and Cleaning Dirty Data. <a href="https://github.com/sfirke/janitor" class="uri">https://github.com/sfirke/janitor</a>. <br />
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. “Welcome to the tidyverse.” Journal of Open Source Software 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686" class="uri">https://doi.org/10.21105/joss.01686</a>. <br />
Wickham, Hadley, Romain François, Lionel Henry, and Kirill Müller. 2021. Dplyr: A Grammar of Data Manipulation. <a href="https://CRAN.R-project.org/package=dplyr" class="uri">https://CRAN.R-project.org/package=dplyr</a>. <br />
Wickham, Hadley, Jim Hester, and Jennifer Bryan. 2021. Readr: Read Rectangular Text Data. <a href="https://CRAN.R-project.org/package=readr" class="uri">https://CRAN.R-project.org/package=readr</a>.</p>
<!--chapter:end:05-data_dimensions.Rmd-->
</div>
</div>
<div id="a-plot-to-choose-from---week-7a" class="section level1 hasAnchor" number="7">
<h1 class="hasAnchor"><span class="header-section-number">7</span> A plot to choose from - Week 7A<a href="#a-plot-to-choose-from---week-7a" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>You should aim to complete this chapter in Week 7, Semester 1</p>
<div id="relationships-differences-and-distributions" class="section level2 hasAnchor" number="7.1">
<h2 class="hasAnchor"><span class="header-section-number">7.1</span> Relationships, differences and distributions<a href="#relationships-differences-and-distributions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As discussed in lectures, the plot you choose to produce should be influenced by the question you are asking.</p>
<p>If you are;</p>
<ul>
<li>Looking at differences between variables - make something like a box plot or bar chart</li>
<li>Looking at relationships between variables - make a scatter plot</li>
<li>Looking at the distribution of data within a variable - make a histogram</li>
</ul>
</div>
<div id="my-first-histogram-in-r" class="section level2 hasAnchor" number="7.2">
<h2 class="hasAnchor"><span class="header-section-number">7.2</span> My first histogram in R<a href="#my-first-histogram-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Lets have a go at making a histogram. Log-in to your posit Cloud account and go to your <code>tribolium_parasites</code> project.</p>
<p>Add the following to your script;</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a histogram and saving it in an object</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>size_histogram <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> wing_case_size)) <span class="sc">+</span> </span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">6</span>) <span class="co"># Tell ggplot that you want it to build a histogram with 6 equal sized bins</span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_histogram) <span class="co"># Print your new figure</span></span></code></pre></div>
<p>Your plot should appear in the plot tab in the bottom right panel of posit Cloud.</p>
<p>A side note on histograms and bins. You may have noticed that we have only specified our x axis variable. This is because R is essentially calculating a frequency for the y axis. How many beetles have a 2.1mm long wing case? How many beetles have a 2.4mm long wing case? This what the y axis is based on. We have asked R to split our x axis variable into 6 equal parts (bins). Have a go at changing the number of bins, what happens to the histogram?</p>
<p>Now have a go at making a histogram for the <code>parasite_burden</code> variable.</p>
<p>Have a look at both of your histograms, what do you think they are showing? We will come back to histograms and how to read them in lectures.</p>
</div>
<div id="exporting-your-histogram" class="section level2 hasAnchor" number="7.3">
<h2 class="hasAnchor"><span class="header-section-number">7.3</span> Exporting your histogram<a href="#exporting-your-histogram" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You are going to want to save your plots, you can use the <code>ggsave</code> function to do this.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving outputs</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;figures/tribolium_wincase_size_histogram.pdf&quot;</span>, <span class="co"># Give R a path to save to and a file name</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> size_histogram, <span class="co"># Tell R what to save - in this case your object</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>) <span class="co"># Tell R what file type to create, in this case a pdf</span></span></code></pre></div>
<p>Note that if you remove the device line <code>ggsave</code> will automatically create a <code>.png</code> file. You can also use <code>width = , height = , units = "cm"</code> arguments to specify the size of your figure in cm. Try to incorporate these new arguments into the above chunk. Save both of your histogram outputs in your <code>figures</code> folder.</p>
</div>
<div id="before-you-leave-2" class="section level2 hasAnchor" number="7.4">
<h2 class="hasAnchor"><span class="header-section-number">7.4</span> Before you leave!<a href="#before-you-leave-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Log out of posit Cloud and make sure you save your script!</p>
</div>
<div id="references-3" class="section level2 hasAnchor" number="7.5">
<h2 class="hasAnchor"><span class="header-section-number">7.5</span> References<a href="#references-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. “Welcome to the tidyverse.” Journal of Open Source Software 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686" class="uri">https://doi.org/10.21105/joss.01686</a>. <br />
Wickham, Hadley, Winston Chang, Lionel Henry, Thomas Lin Pedersen, Kohske Takahashi, Claus Wilke, Kara Woo, Hiroaki Yutani, and Dewey Dunnington. 2021. Ggplot2: Create Elegant Data Visualisations Using the Grammar of Graphics. <a href="https://CRAN.R-project.org/package=ggplot2" class="uri">https://CRAN.R-project.org/package=ggplot2</a>.</p>
<!--chapter:end:06-a_plot_to_choose_from.Rmd-->
</div>
</div>
<div id="lost-the-plot-some-more-data-visualisations---week-8a---workshop-2" class="section level1 hasAnchor" number="8">
<h1 class="hasAnchor"><span class="header-section-number">8</span> Lost the plot? Some more data visualisations - Week 8A - Workshop 2<a href="#lost-the-plot-some-more-data-visualisations---week-8a---workshop-2" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This is our second workshop hopefully you are gaining some confidence with how the underlying functionality of R works. For this workshop we will be building plots to explore possible relationships and differences within the data. Don’t forget to ask for help if you are stuck on any of the material covered so far.</p>
<div id="looking-at-differences" class="section level2 hasAnchor" number="8.1">
<h2 class="hasAnchor"><span class="header-section-number">8.1</span> Looking at differences<a href="#looking-at-differences" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As we discussed in the Chapter 4 wing case length is a good proxy for overall beetle size in <em>Tribolium</em>. So we will be comparing beetle wing case length in male and female beetles. Because we are looking for possible differences we know that we probably want to build a box plot. Login to your posit Cloud account and copy over the following chunk of code into your <code>tribolium_parasites.r</code> script. Run it and see what happens.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a box plot and saving it in an object</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis and y axis</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>size_differences_by_sex_boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> sex, y wing_case_size)) <span class="sc">+</span> </span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="co"># Tell ggplot that you want it to build a boxplot</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_differences_by_sex_boxplot) <span class="co"># Print your new figure</span></span></code></pre></div>
<p>Oh dear… Looks like there may be something wrong with the script…</p>
<pre><code>geom_boxplot: outlier.colour = NULL, outlier.fill = NULL, outlier.shape = 19, outlier.size = 1.5, outlier.stroke = 0.5, outlier.alpha = NULL, notch = FALSE, notchwidth = 0.5, varwidth = FALSE, na.rm = FALSE, orientation = NA
stat_boxplot: na.rm = FALSE, orientation = NA
position_dodge2 </code></pre>
<p>When you get errors like this it generally means that ggplot is missing something essential from the script. There is a typo in the original script, can you spot it and fix it? Once you have made the figure, take a look at it, do you think that there is a difference in size between male and female beetles? How would you interpret this figure? Make sure you save your box plot to your figures folder.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving outputs</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;figures/size_differences_by_sex_boxplot.pdf&quot;</span>, <span class="co"># Give R a path to save to and a file name</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> size_differences_by_sex_boxplot, <span class="co"># Tell R what to save - in this case your object</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>) <span class="co"># Tell R what file type to create, in this case a pdf</span></span></code></pre></div>
<p>Our other categorical variable is treatment. Using the chunk of code within your script try to write a new chunk to compare the differences between <code>wing_case_size</code> and <code>treatment</code> in our sample of <em>Tribolium</em>.</p>
<p>Oh dear… R still isn’t happy… You have probably been presented with the following message</p>
<pre><code>Warning message:
Continuous x aesthetic -- did you forget aes(group=...)? </code></pre>
<p>This essentially means that R is expecting a categorical variable (e.g. blue, red, green or as our example is 30 or 35 degrees centigrade) but what it sees is a continuous variable (e.g. 1,2,3,4,5,6,7,8,9). This has come about because although our treatment is categorical it is expressed as a numeric and has confused R. Use the <code>glimpse()</code> command to see how R has interpreted the <code>treatment</code> variable. This should confirm that R sees <code>treatment</code> as a double (<code>dbl</code>) vector. We missed this when we were checking our data in Chapter 4, but thankfully it is an easy fix. We want to tell R to treat the variable <code>treatment</code> as a factor (categorical data) and not as continuous data. Try adding this to your script;</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Instruct R to treat the treatment variable as a factor </span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>tribolium_combo<span class="sc">$</span>treatment <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(tribolium_combo<span class="sc">$</span>treatment) <span class="co"># Note this will edit your data frame</span></span></code></pre></div>
<p>Now run <code>glimpse()</code> again. Can you spot the difference?</p>
<p>Try to make your box plot for <code>wing_case_size</code> and <code>treatment</code> again. What does this plot tell you about the data set? Make sure you save it to your <code>figures</code> folder.</p>
<p>It would be really useful to show both <code>sex</code> and <code>treatment</code> on the x axis so that you can compare between both categorical variables and size. We can do this with a fairly simple addition to our chunk. Add the expression <code>fill = sex</code> to the <code>aes</code> function of your last chunk. This tells R to colour your box plot by sex. You should therefore have 4 boxes. Save this figure under a different name in your <code>figures folder</code>.</p>
<p>You should have five figures (2 histogram and 3 box plots) saved there now. Look over your box plots, can you see how your interpretation of the data might change depending on which plot you look at? Which plot gives you the most complete picture?</p>
<p>We will discus this further in lectures. But for now lets have a look at visualising a relationship.</p>
</div>
<div id="looking-at-relationships" class="section level2 hasAnchor" number="8.2">
<h2 class="hasAnchor"><span class="header-section-number">8.2</span> Looking at relationships<a href="#looking-at-relationships" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To look for possible relationships we need two continuous variables. In our data set we have <code>wing_case_size</code> and <code>parasite_burden</code>. There are several theories in the field of parasitology that suggest larger organisms tend to harbor more parasites. We can see if this is a relationship that persists in our <em>Tribolium</em>. Copy, complete and run the following chunk. You may notice that x and y axis variables have been left blank, which variable do you think belongs on which axis?</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a scatter plot and saving it in an object</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis and y axis</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>size_vs_parasites_point <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> , <span class="at">y =</span> )) <span class="sc">+</span> </span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="co"># geom_point is ggplots scatter plot</span></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_vs_parasites_point)</span></code></pre></div>
<p>What do you think this plot is showing you? Try adding another function to this chunk, add <code>geom_smooth(method="lm")</code>. This will add a simple regression line to your plot, I suggest you add a suitable comment to that effect to your chunk. Does this new addition to your plot support any initial observations you may have made from the plot? Make sure you save your plot in your <code>figures</code> folder.</p>
<p>Well done, you have started building two very commonly used plots using R. These plots are functionally sound, but lets be honest, they’re not pretty. A plot should always be pretty. Next week we will start learning how to make plots visually pleasing as well as functional.</p>
</div>
<div id="before-you-leave-3" class="section level2 hasAnchor" number="8.3">
<h2 class="hasAnchor"><span class="header-section-number">8.3</span> Before you leave!<a href="#before-you-leave-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Log out of posit Cloud and make sure you save your script!</p>
</div>
<div id="references-4" class="section level2 hasAnchor" number="8.4">
<h2 class="hasAnchor"><span class="header-section-number">8.4</span> References<a href="#references-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. “Welcome to the tidyverse.” Journal of Open Source Software 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686" class="uri">https://doi.org/10.21105/joss.01686</a>. <br />
Wickham, Hadley, Winston Chang, Lionel Henry, Thomas Lin Pedersen, Kohske Takahashi, Claus Wilke, Kara Woo, Hiroaki Yutani, and Dewey Dunnington. 2021. Ggplot2: Create Elegant Data Visualisations Using the Grammar of Graphics. <a href="https://CRAN.R-project.org/package=ggplot2" class="uri">https://CRAN.R-project.org/package=ggplot2</a>.</p>
<!--chapter:end:07-lost_the_plot.Rmd-->
</div>
</div>
<div id="make-them-pretty---week-9a" class="section level1 hasAnchor" number="9">
<h1 class="hasAnchor"><span class="header-section-number">9</span> Make them pretty - Week 9A<a href="#make-them-pretty---week-9a" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>You should aim to complete this chapter in Week 9, Semester 1</p>
<p>Over the last couple of weeks you have build histograms, box plots and scatter plots. Probably the three types of data visualisation you are most likely to run into and want to build. But you may have noticed that non of these are particularly aesthetically pleasing. As a rule, data visualisations should always be designed to be attractive, clean, clear and nicely spaced. The purpose of a plot is to provide, you and your audience with a accurate, visually pleasing and easy to interpret representation of your data. Thankfully <code>ggplot2</code> contains all the tools we need to do this.</p>
<p>Open your <code>tribolium_parasites</code> project in posit Cloud. Create a new script and call it <code>ggplot_fundamentals</code>, save it in your <code>scripts</code> folder and set up your script following the rules in Chapter 4, make sure the <code>tidyverse</code> library is loaded, this contains <code>ggplot2</code>.</p>
<p>Sometimes its can be good to cleanup your <strong>Environment</strong> a bit, if you run the following command, it will delete everything stored in your environment. Note - you will have to reload your data frames from your data folder. Make sure you saved <code>tribolium_combo</code> to your data folder in Chapter 4 before you do this.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up your environment</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code></pre></div>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Reload your tribolium_combo data frame</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>tribolium_combo <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/tribolium_combo.csv&quot;</span>)</span></code></pre></div>
<p>Check your data has loaded correctly. Can you remember how to change <code>treatment</code> from a <code>dbl</code> vector to a factor (<code>fct</code>)?</p>
<div id="deconstructing-ggplot" class="section level2 hasAnchor" number="9.1">
<h2 class="hasAnchor"><span class="header-section-number">9.1</span> Deconstructing ggplot<a href="#deconstructing-ggplot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To get an idea of how ggplot works we are going to go back over some code you have already run. Try running the following, what happens?</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a box plot and saving it in an object</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis and y axis</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>size_differences_boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> sex, <span class="at">y =</span> wing_case_size)) </span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_differences_boxplot) <span class="co"># Print the object your plot is stored in to view it</span></span></code></pre></div>
<p>So you have given ggplot your data set and specified the axes. But you haven’t told it how you would like the data to be visualised. So you should see something like this…</p>
<div class="figure">
<img src="figures/screenshot_deconstructingaplot_1.png" alt="Basic ggplot output" width="90%" />
<p class="caption">
(#fig:unnamed-chunk-49)The most basic output from ggplot
</p>
</div>
<p>Now modify the script so that it looks like this…</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a box plot and saving it in an object</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis and y axis</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>size_differences_boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> sex, <span class="at">y =</span> wing_case_size)) <span class="sc">+</span> </span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> treatment)) <span class="co"># Tell ggplot that you want it to build a box plot coloured by treatment</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_differences_boxplot) <span class="co"># Print the object your plot is stored in to view it</span></span></code></pre></div>
<p>Note the <code>+</code>, this is essentially another way of piping information from one function into the next. You could also have added <code>fill = treatment</code> to the first line within this chunk within <code>aes()</code>.</p>
<p>Your figure should look like this…</p>
<div class="figure">
<img src="figures/screenshot_deconstructingaplot_2.png" alt="Basic ggplot boxplot" width="90%" />
<p class="caption">
(#fig:unnamed-chunk-51)A basic ggplot box plot output
</p>
</div>
<p>Have a look at this figure, is there anything you would change, to make it more visually appealing or clear?</p>
<p>Here are some things I would change:</p>
<ul>
<li>Labels</li>
<li>Colour scheme</li>
<li>Spacing</li>
<li>Theme</li>
</ul>
<p>We will look into labels and colour scheme this week and spacing and theme next week.</p>
</div>
<div id="labels" class="section level2 hasAnchor" number="9.2">
<h2 class="hasAnchor"><span class="header-section-number">9.2</span> Labels<a href="#labels" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Clear accurate labeling is essential in the sciences, be it when your in the lab labeling up your samples or creating data visualisations.</p>
<p>In our current plot both our x and y axis could do with some relabeling, just because we dont use capitals when coding doesn’t mean we don’t follow the normal rules of English grammar when presenting data. Try adjusting your code so that it looks like the following;</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a box plot and saving it in an object</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis and y axis</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>size_differences_boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> sex, <span class="at">y =</span> wing_case_size)) <span class="sc">+</span> </span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> treatment)) <span class="sc">+</span> <span class="co"># Tell ggplot that you want it to build a boxplot</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Sex&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Wing case size (mm)</span><span class="sc">\n</span><span class="st">&quot;</span>) <span class="co"># Adjust your x and y axis labels </span></span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_differences_boxplot) <span class="co"># Print the object your plot is stored in to view it</span></span></code></pre></div>
<p>Note the <code>\n</code> on your y axis label. This simply adds a new line to the label and spaces it nicely away from the y axis.</p>
<p>This looks a bit better but our legend is still not very well labeled. Adjust your <code>labs</code> function so that it reads <code>labs(x = "Sex", y = "Wing case size (mm)\n", fill = "Treatment")</code>. This will change the label for your legend.</p>
<p>Our labels still aren’t quite right, I don’t like the abbreviations on the x axis and our legend could still use some work as there are no units given for the treatment temperatures. Try these changes;</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a box plot and saving it in an object</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis and y axis</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>size_differences_boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> sex, <span class="at">y =</span> wing_case_size)) <span class="sc">+</span> </span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> treatment)) <span class="sc">+</span> <span class="co"># Tell ggplot that you want it to build a boxplot</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Sex&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Wing case size (mm)</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;Treatment&quot;</span>) <span class="sc">+</span> <span class="co"># Adjust your legend and x and y axis labels </span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>,<span class="st">&quot;Male&quot;</span>)) <span class="sc">+</span> <span class="co"># Rename the categories on the x axis </span></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;30°C&quot;</span>, <span class="st">&quot;35°C&quot;</span>)) <span class="co"># Rename your treatments</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_differences_boxplot) <span class="co"># Print the object your plot is stored in to view it</span></span></code></pre></div>
<p>Note that the <code>scale x discrete</code> and <code>scale_fill_manual</code> functions simply rename things in the same order as you present them, make sure you label your categories accurately or this can make a big mess later on.</p>
<p>If you try to remake your plot at this stage you will get the following error:</p>
<pre><code>Error in `f()`:
! Insufficient values in manual scale. 2 needed but only 0 provided.
Run `rlang::last_error()` to see where the error occurred.</code></pre>
<p>This is because <code>scale_fill_manual</code> is also expecting some instructions on how to colour your boxplot. You will need to complete section 9.3 before you can successfully remake your boxplot.</p>
</div>
<div id="switching-up-colours" class="section level2 hasAnchor" number="9.3">
<h2 class="hasAnchor"><span class="header-section-number">9.3</span> Switching up colours<a href="#switching-up-colours" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The colours of your boxes are the default colours given by <code>ggplot2</code>. We can modify these to make our plot look a bit more pleasing. Try editing the <code>scale_fill_manual</code> function to this <code>scale_fill_manual(labels = c("30°C", "35°C"), values = c("cornflowerblue", "coral"))</code>. Run it and see what happens to your plot.</p>
<p>What do you think of your new plot now? Do you think the changes to labels and colours are an improvement?</p>
<p>R has lots of <a href="https://www.datanovia.com/en/blog/awesome-list-of-657-r-color-names/">colours</a>, take a look at the linked reference lists and have a play with changing up some of your colours.</p>
<p>Colour can be a really useful tool to employ when making your plots visually appealing, however make sure you are mindful that some colour pallets can be difficult for some people to interpret. There are however, some really good <a href="https://www.tableau.com/en-gb/about/blog/examining-data-viz-rules-dont-use-red-green-together#:~:text=Use%20a%20colour%2Dblind%2Dfriendly%20palette%20when%20appropriate&amp;text=For%20example%2C%20blue%2Forange%20is,blue%20to%20someone%20with%20CVD">tips</a> out there for making sure your figures are accessible to everyone.</p>
</div>
<div id="before-you-leave-4" class="section level2 hasAnchor" number="9.4">
<h2 class="hasAnchor"><span class="header-section-number">9.4</span> Before you leave!<a href="#before-you-leave-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Log out of posit Cloud and make sure you save your script!</p>
</div>
<div id="references-5" class="section level2 hasAnchor" number="9.5">
<h2 class="hasAnchor"><span class="header-section-number">9.5</span> References<a href="#references-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. “Welcome to the tidyverse.” Journal of Open Source Software 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686" class="uri">https://doi.org/10.21105/joss.01686</a>. <br />
Wickham, Hadley, Winston Chang, Lionel Henry, Thomas Lin Pedersen, Kohske Takahashi, Claus Wilke, Kara Woo, Hiroaki Yutani, and Dewey Dunnington. 2021. Ggplot2: Create Elegant Data Visualisations Using the Grammar of Graphics. <a href="https://CRAN.R-project.org/package=ggplot2" class="uri">https://CRAN.R-project.org/package=ggplot2</a>.</p>
<!--chapter:end:08-make_them_pretty.Rmd-->
</div>
</div>
<div id="make-them-pretty-continued---week-10a" class="section level1 hasAnchor" number="10">
<h1 class="hasAnchor"><span class="header-section-number">10</span> Make them pretty continued - Week 10A<a href="#make-them-pretty-continued---week-10a" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>You should aim to complete this chapter in Week 10, Semester 1</p>
<p>In Chapter 9 we explored the fundamental functionality of <code>ggplot2</code> and we started to play with some functions that improved our plots labels and colour scheme. This week we will be extending this further and investigating how we may also be able to alter spacing and themes to improve the appearance of our plots.</p>
<p>Open your <code>tribolium_parasites</code> project in posit Cloud and your <code>ggplot_fundamentals.r</code> script.</p>
<p>Your current figure chunk should look something like this;</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a box plot and saving it in an object</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis and y axis</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>size_differences_boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> sex, <span class="at">y =</span> wing_case_size)) <span class="sc">+</span> </span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> treatment)) <span class="sc">+</span> <span class="co"># Tell ggplot that you want it to build a box plot filled according to treatment</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Sex&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Wing case size (mm)</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;Treatment&quot;</span>) <span class="sc">+</span> <span class="co"># Adjust your legend and x and y axis labels </span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>,<span class="st">&quot;Male&quot;</span>)) <span class="sc">+</span> <span class="co"># Rename the categories on the x axis </span></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;30°C&quot;</span>, <span class="st">&quot;35°C&quot;</span>), <span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;cornflowerblue&quot;</span>, <span class="st">&quot;coral&quot;</span>)) <span class="co"># Rename and colour your treatments</span></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_differences_boxplot) <span class="co"># Print the object your plot is stored in to view it</span></span></code></pre></div>
<div id="spacing" class="section level2 hasAnchor" number="10.1">
<h2 class="hasAnchor"><span class="header-section-number">10.1</span> Spacing<a href="#spacing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Lets have a look at spacing. This is a box plot so we the main spacing option you are likely to want to play with is the width of your boxes. Here we can simply add an argument to the <code>geom_boxplot</code> function. Edit this function so that it reads <code>geom_boxplot(aes(fill = treatment), width = 0.9)</code>. The width argument can be anything between 0.00 or 1.00. It changes the width of the boxes and this changes the spacing between them as well. Have a play with some values and see what happens.</p>
<p>We will come back to spacing when we look at our scatter plots next week.</p>
</div>
<div id="themes" class="section level2 hasAnchor" number="10.2">
<h2 class="hasAnchor"><span class="header-section-number">10.2</span> Themes<a href="#themes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>So the final aspect of aesthetics we are going to look at here is themes. We have a reasonably attractive graph now, but its still got a grey background and the grid lines are unnecessary. To remove the grey background and implement the classic black on white aesthetic we can simply add a function that defines a pre-made theme. Adjust your script so that it includes the function <code>theme_bw()</code>, I suggest you add this as a new line, don’t forget to pipe between your functions with a <code>+</code>. Run this chunk and print your new plot. How is that looking now?</p>
<p>Two things still jump out at me when looking at this plot. The grid lines are completely unnecessary and detract from the overall aesthetic and the text sizes could be larger. To make these edits we can simply add additional instructions to adjust the theme further. So although most of the work has been done by applying a the <code>theme_bw</code> we still need make some adjustments.</p>
<p>Add the following function to your growing ggplot chunk (don’t forget to pipe <code>+</code> between functions);</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">theme</span>(<span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">color=</span><span class="st">&quot;black&quot;</span>), <span class="co"># Specifies that the plot boarder is coloured black</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes minor grid lines </span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>()) <span class="co"># Removes major grid lines </span></span></code></pre></div>
<p>Now we just need to adjust the text size. We can do this within the <code>theme</code> function as well adjust your theme so that it reads like this;</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">theme</span>(<span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">color=</span><span class="st">&quot;black&quot;</span>), <span class="co"># Specifies that the plot boarder is coloured black</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes minor grid lines </span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes major grid lines </span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>), <span class="co"># Changes the size of text on both axis </span></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>), <span class="co"># Changes size of your axis labels </span></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>), <span class="co"># Changes the size of text within your legend</span></span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>)) <span class="co"># Changes the size of the legend title</span></span></code></pre></div>
<p>Have a play with the different text sizes until you think they are optimal.</p>
<p>Hopefully you now have a nice clean, clear and aesthetically pleasing plot and have some awareness of the commands and functions used to make it. Next week is our last workshop of the term and we will be applying these rules to all of the plots made in the previous chapters. This will also give you the oppotunity to ask any questions.</p>
</div>
<div id="before-you-leave-5" class="section level2 hasAnchor" number="10.3">
<h2 class="hasAnchor"><span class="header-section-number">10.3</span> Before you leave!<a href="#before-you-leave-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Log out of posit Cloud and make sure you save your script!</p>
</div>
<div id="references-6" class="section level2 hasAnchor" number="10.4">
<h2 class="hasAnchor"><span class="header-section-number">10.4</span> References<a href="#references-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. “Welcome to the tidyverse.” Journal of Open Source Software 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686" class="uri">https://doi.org/10.21105/joss.01686</a>. <br />
Wickham, Hadley, Winston Chang, Lionel Henry, Thomas Lin Pedersen, Kohske Takahashi, Claus Wilke, Kara Woo, Hiroaki Yutani, and Dewey Dunnington. 2021. Ggplot2: Create Elegant Data Visualisations Using the Grammar of Graphics. <a href="https://CRAN.R-project.org/package=ggplot2" class="uri">https://CRAN.R-project.org/package=ggplot2</a>.</p>
<!--chapter:end:09-make_them_pretty_continued.Rmd-->
</div>
</div>
<div id="the-subtle-art-of-cannibalisation---week-11a---workshop-3" class="section level1 hasAnchor" number="11">
<h1 class="hasAnchor"><span class="header-section-number">11</span> The subtle art of cannibalisation - Week 11A - Workshop 3<a href="#the-subtle-art-of-cannibalisation---week-11a---workshop-3" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this Week 11 workshop we will be applying the rules of aesthetics that we learned in the last two chapters to the rest of our plots. You may use functions and arguments from your <code>ggplot_fundamentals.r</code> script to modify the plots you made in previous weeks. This is a very common practice, especially while learning a new programming language, you get a script or chunk of code right once and then canibalise parts of it for other uses, modifying it as needed to fit its new purpose.</p>
<div id="starting-on-the-right-footing" class="section level2 hasAnchor" number="11.1">
<h2 class="hasAnchor"><span class="header-section-number">11.1</span> Starting on the right footing<a href="#starting-on-the-right-footing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In the last few chapters, and over the last few weeks you have been asked to modify quite a few chunks of script. If anything so far hasn’t made sense or you haven’t managed to get some chunks or functions working please do let either myself or a demonstrator know before you start the workshop activity. We are here to help!</p>
</div>
<div id="beautifying-your-box-plots" class="section level2 hasAnchor" number="11.2">
<h2 class="hasAnchor"><span class="header-section-number">11.2</span> Beautifying your box plots<a href="#beautifying-your-box-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In Chapters 9 and 10 we specifically worked on making changes to our box plots that would make them clean, clear and visually pleasing. You have all the code already made for your box plots, it just needs tailoring a little. You should have three boxplots saved in your <code>figures</code> folder;</p>
<ul>
<li>size_differences_by_sex_boxplot.pdf</li>
<li>size_differences_by_treatment_boxplot.pdf</li>
<li>size_differences_boxplot.pdf - remember this is filled by <code>sex</code> not <code>treatment</code> so will be subtly different to the figure made in the <code>ggplot_fundamentals.R</code> script that we worked on in Chapters 9 and 10.</li>
</ul>
<p>Using the script from <code>ggplot_fundamentals.R</code> make the necessary modifications to each of these three figures so that they are clean, clear and visually pleasing. Make sure you save each one to your <code>figures</code> folder using <code>ggsave</code>.</p>
</div>
<div id="beautifying-your-histograms" class="section level2 hasAnchor" number="11.3">
<h2 class="hasAnchor"><span class="header-section-number">11.3</span> Beautifying your histograms<a href="#beautifying-your-histograms" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You hopefully also have two histograms saved in your <code>figures</code> folder;</p>
<ul>
<li>size_histogram.pdf</li>
<li>parasite_histogram.pdf</li>
</ul>
<p>Use you previous chunks of scripts and knowledge of R to edit the labels and themes of this plot. Print it again and take a look. When you’re happy save your updated figures to the <code>figures</code> folder using <code>ggsave</code>.</p>
</div>
<div id="beautifying-your-scatter-plot" class="section level2 hasAnchor" number="11.4">
<h2 class="hasAnchor"><span class="header-section-number">11.4</span> Beautifying your scatter plot<a href="#beautifying-your-scatter-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You hopefully also have a scatter plot saved in your <code>figures</code> folder;</p>
<ul>
<li>size_vs_parasites_point.pdf</li>
</ul>
<p>Use you previous chunks of scripts and knowledge of R to edit the labels and themes of this plot. Print it again and take a look.</p>
<p>Now there are some additional elements in this plot that could be adjusted. These include;</p>
<ul>
<li>Point shape - The shape of the individual points on the plot, as with colours there are lots of options numbered 0 - 25, descriptions of each one are listed <a href="https://www.datanovia.com/en/blog/ggplot-point-shapes-best-tips/">here</a></li>
<li>Point size - The size of each point</li>
<li>Point colour - The colour of your points</li>
<li>Line type (solid, dashed, etc) - Again thre are several options here, the notation and descriptions of which can be found <a href="http://www.sthda.com/english/wiki/ggplot2-line-types-how-to-change-line-types-of-a-graph-in-r-software">here</a></li>
<li>Line colour</li>
<li>Line size - the weight of the line</li>
<li>Line fill - Note that here this refers to the colour of the shaded area marking the standard error</li>
</ul>
<p>So lets play with some of these. Try adding and adjusting the following arguments within <code>geom_point()</code></p>
<ul>
<li><code>shape = 1</code> - have a look at other point shapes you could use and play with this setting</li>
<li><code>size = 2</code> - again try playing with some point sizes</li>
<li><code>colour = "blue"</code> - or any other <a href="https://www.datanovia.com/en/blog/awesome-list-of-657-r-color-names/">colour</a> you fancy trying.</li>
</ul>
<p>Once you are happy with your points we can take a look at your regression line. Try adding and adjusting the following arguments within <code>geom_smooth()</code></p>
<ul>
<li><code>colour = "cornflowerblue"</code></li>
<li><code>fill = "lightblue"</code></li>
<li><code>size = 1</code></li>
<li><code>linetype = "dashed"</code></li>
</ul>
<p>As before please do experiment and play with the settings described by each of these arguments.</p>
<p>Once you are happy with your new and beautiful scatter plot make sure you save it to to the <code>figures</code> folder using <code>ggsave</code>.</p>
</div>
<div id="a-challenge" class="section level2 hasAnchor" number="11.5">
<h2 class="hasAnchor"><span class="header-section-number">11.5</span> A Challenge!<a href="#a-challenge" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Try to adjust the chunk for your size vs parasites scatter plot so that the points are coloured by <code>sex</code> and shaped by <code>treatment</code>. Colour female points <code>"deepskyblue3"</code> and male points <code>"coral"</code> and use solid squares for the 30°C treatment and solid circles for the 35°C treatment. Make sure the text and titles for your legends are correct and well presented.</p>
<p>Hint - to do this you will need to correctly use and populate the <code>aes()</code> function in <code>geom_point()</code> and then to adjust colours, shapes and associated text you will need to pipe <code>+</code> to the <code>scale_color_manual()</code> and the <code>scale_shape_manual()</code> functions. You will also need to adjust the settings in the <code>labs()</code> function to change the legend titles.</p>
</div>
<div id="end-of-semester-1" class="section level2 hasAnchor" number="11.6">
<h2 class="hasAnchor"><span class="header-section-number">11.6</span> End of Semester 1<a href="#end-of-semester-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This is the last of the taught content for the Introduction to Data Science sub module. You should all be very proud of yourselves for completing the first half of this workbook! Don’t forget to complete the two blackboard quizzes before your Skills Essay deadline and I wish you all a lovely Christmas break!</p>
<p>See you all in the new year :)</p>
</div>
<div id="before-you-leave-6" class="section level2 hasAnchor" number="11.7">
<h2 class="hasAnchor"><span class="header-section-number">11.7</span> Before you leave!<a href="#before-you-leave-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Log out of posit Cloud and make sure you save your script!</p>
</div>
<div id="references-7" class="section level2 hasAnchor" number="11.8">
<h2 class="hasAnchor"><span class="header-section-number">11.8</span> References<a href="#references-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. “Welcome to the tidyverse.” Journal of Open Source Software 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686" class="uri">https://doi.org/10.21105/joss.01686</a>. <br />
Wickham, Hadley, Winston Chang, Lionel Henry, Thomas Lin Pedersen, Kohske Takahashi, Claus Wilke, Kara Woo, Hiroaki Yutani, and Dewey Dunnington. 2021. Ggplot2: Create Elegant Data Visualisations Using the Grammar of Graphics. <a href="https://CRAN.R-project.org/package=ggplot2" class="uri">https://CRAN.R-project.org/package=ggplot2</a>.</p>
<!--chapter:end:10-cannibalisation.Rmd-->
</div>
</div>
<div id="merry-christmas" class="section level1 hasAnchor" number="12">
<h1 class="hasAnchor"><span class="header-section-number">12</span> Merry Christmas!!!<a href="#merry-christmas" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<script>
  addClassKlippyTo("pre.r, pre.bash");
  addKlippy('right', 'top', 'auto', '1', 'copy to clipboard', 'Copied!');
</script>
<p>You have all done fanatically well in semester A so I have got you all a lovely Christmas gift… Fully commented and completed versions of the scripts you have been working on. I would still suggest you try to complete the workbook in your own time, this will mean that you have a much better understanding of how the code works and will be better for you in the long run. But these commented and completed scripts may give you something to compare your own work too and will hopefully allow us all to start on the same page in the New Year.</p>
<details>
<summary>
<strong>Show me the Tribolium parasites script</strong>
</summary>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co"># An analysis of parasite abundance between male and female tribolium beetles from two thermal treatments (30 and 35 degrees centigrade)</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Ellen Bell </span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 08/08/2022</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse) <span class="co"># Loads the tidyverse package</span></span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(janitor) <span class="co"># Loads the janitor package</span></span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the data from the .csv file in the data folder into my project</span></span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>tribolium <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/Tribolium_parasites.csv&quot;</span>) <span class="co">#</span></span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tribolium)</span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-13"><a href="#cb55-13" aria-hidden="true" tabindex="-1"></a><span class="co"># First of all lets create some variables</span></span>
<span id="cb55-14"><a href="#cb55-14" aria-hidden="true" tabindex="-1"></a>beetle_id <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>) <span class="co"># Notice that this notation will create of a list of numbers 1 to 20</span></span>
<span id="cb55-15"><a href="#cb55-15" aria-hidden="true" tabindex="-1"></a>sex <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;F&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;M&quot;</span>)</span>
<span id="cb55-16"><a href="#cb55-16" aria-hidden="true" tabindex="-1"></a>treatment <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">35</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">30</span>)</span>
<span id="cb55-17"><a href="#cb55-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-18"><a href="#cb55-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Turning variables into a tibble</span></span>
<span id="cb55-19"><a href="#cb55-19" aria-hidden="true" tabindex="-1"></a>tribolium_extras <span class="ot">&lt;-</span> <span class="fu">tibble</span>(beetle_id,sex,treatment)</span>
<span id="cb55-20"><a href="#cb55-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-21"><a href="#cb55-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Renaming some variables in your original data set</span></span>
<span id="cb55-22"><a href="#cb55-22" aria-hidden="true" tabindex="-1"></a>tribolium <span class="ot">&lt;-</span> tribolium <span class="sc">%&gt;%</span></span>
<span id="cb55-23"><a href="#cb55-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">beetle_id =</span> beetle_id)</span>
<span id="cb55-24"><a href="#cb55-24" aria-hidden="true" tabindex="-1"></a>tribolium <span class="ot">&lt;-</span> tribolium <span class="sc">%&gt;%</span></span>
<span id="cb55-25"><a href="#cb55-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">wing_case_size =</span> wing_case_size)</span>
<span id="cb55-26"><a href="#cb55-26" aria-hidden="true" tabindex="-1"></a>tribolium <span class="ot">&lt;-</span> tribolium <span class="sc">%&gt;%</span></span>
<span id="cb55-27"><a href="#cb55-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">parasite_burden =</span> parasite_burden)</span>
<span id="cb55-28"><a href="#cb55-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-29"><a href="#cb55-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Check how your data frame now looks</span></span>
<span id="cb55-30"><a href="#cb55-30" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tribolium)</span>
<span id="cb55-31"><a href="#cb55-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-32"><a href="#cb55-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge data frames</span></span>
<span id="cb55-33"><a href="#cb55-33" aria-hidden="true" tabindex="-1"></a>tribolium_combo <span class="ot">&lt;-</span> <span class="fu">merge</span>(tribolium, tribolium_extras, <span class="at">by =</span> <span class="st">&quot;beetle_id&quot;</span>) <span class="co"># This merge function combines both of your data frames but matches rows by beetle_id</span></span>
<span id="cb55-34"><a href="#cb55-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-35"><a href="#cb55-35" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(tribolium_combo)</span>
<span id="cb55-36"><a href="#cb55-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-37"><a href="#cb55-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Save your combined data frame as a .csv file (which you could download) to your data folder</span></span>
<span id="cb55-38"><a href="#cb55-38" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(tribolium_combo,<span class="st">&quot;data/tribolium_combo.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb55-39"><a href="#cb55-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-40"><a href="#cb55-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the n_distinct function to count unique/distinct entries in the beetle_id column</span></span>
<span id="cb55-41"><a href="#cb55-41" aria-hidden="true" tabindex="-1"></a><span class="fu">n_distinct</span>(tribolium_combo<span class="sc">$</span>beetle_id)</span>
<span id="cb55-42"><a href="#cb55-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-43"><a href="#cb55-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the tabyl function from the janitor package, comment on what it does. </span></span>
<span id="cb55-44"><a href="#cb55-44" aria-hidden="true" tabindex="-1"></a><span class="fu">tabyl</span>(tribolium_combo, sex , treatment)</span>
<span id="cb55-45"><a href="#cb55-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-46"><a href="#cb55-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a histogram and saving it in an object</span></span>
<span id="cb55-47"><a href="#cb55-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis</span></span>
<span id="cb55-48"><a href="#cb55-48" aria-hidden="true" tabindex="-1"></a>size_histogram <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> wing_case_size)) <span class="sc">+</span> </span>
<span id="cb55-49"><a href="#cb55-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">6</span>) <span class="co"># Tell ggplot that you want it to build a histogram with 6 equal sized bins</span></span>
<span id="cb55-50"><a href="#cb55-50" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_histogram) <span class="co"># Print your new figure</span></span>
<span id="cb55-51"><a href="#cb55-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-52"><a href="#cb55-52" aria-hidden="true" tabindex="-1"></a>parasite_burden_histogram <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> parasite_burden)) <span class="sc">+</span> </span>
<span id="cb55-53"><a href="#cb55-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">6</span>) <span class="co"># Tell ggplot that you want it to build a histogram with 6 equal sized bins</span></span>
<span id="cb55-54"><a href="#cb55-54" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(parasite_burden_histogram) <span class="co"># Print your new figure</span></span>
<span id="cb55-55"><a href="#cb55-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-56"><a href="#cb55-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving outputs</span></span>
<span id="cb55-57"><a href="#cb55-57" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;figures/tribolium_wingcase_size_histogram.pdf&quot;</span>, <span class="co"># Give R a path to save to and a file name</span></span>
<span id="cb55-58"><a href="#cb55-58" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> size_histogram, <span class="co"># Tell R what to save - in this case your object</span></span>
<span id="cb55-59"><a href="#cb55-59" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>) <span class="co"># Tell R what file type to create, in this case a pdf</span></span>
<span id="cb55-60"><a href="#cb55-60" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;figures/tribolium_parasite_burden_histogram.pdf&quot;</span>, <span class="co"># Give R a path to save to and a file name</span></span>
<span id="cb55-61"><a href="#cb55-61" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> parasite_burden_histogram, <span class="co"># Tell R what to save - in this case your object</span></span>
<span id="cb55-62"><a href="#cb55-62" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>) <span class="co"># Tell R what file type to create, in this case a pdf</span></span>
<span id="cb55-63"><a href="#cb55-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-64"><a href="#cb55-64" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a box plot and saving it in an object</span></span>
<span id="cb55-65"><a href="#cb55-65" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis and y axis</span></span>
<span id="cb55-66"><a href="#cb55-66" aria-hidden="true" tabindex="-1"></a>size_differences_by_sex_boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> sex, <span class="at">y =</span> wing_case_size)) <span class="sc">+</span> </span>
<span id="cb55-67"><a href="#cb55-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="co"># Tell ggplot that you want it to build a boxplot</span></span>
<span id="cb55-68"><a href="#cb55-68" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_differences_by_sex_boxplot) <span class="co"># Print your new figure</span></span>
<span id="cb55-69"><a href="#cb55-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-70"><a href="#cb55-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving outputs</span></span>
<span id="cb55-71"><a href="#cb55-71" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;figures/size_differences_by_sex_boxplot.pdf&quot;</span>, <span class="co"># Give R a path to save to and a file name</span></span>
<span id="cb55-72"><a href="#cb55-72" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> size_differences_by_sex_boxplot, <span class="co"># Tell R what to save - in this case your object</span></span>
<span id="cb55-73"><a href="#cb55-73" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>) <span class="co"># Tell R what file type to create, in this case a pdf</span></span>
<span id="cb55-74"><a href="#cb55-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-75"><a href="#cb55-75" aria-hidden="true" tabindex="-1"></a><span class="co"># Instruct R to treat the treatment variable as a factor </span></span>
<span id="cb55-76"><a href="#cb55-76" aria-hidden="true" tabindex="-1"></a>tribolium_combo<span class="sc">$</span>treatment <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(tribolium_combo<span class="sc">$</span>treatment) <span class="co"># Note this will edit your data frame</span></span>
<span id="cb55-77"><a href="#cb55-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-78"><a href="#cb55-78" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a box plot and saving it in an object</span></span>
<span id="cb55-79"><a href="#cb55-79" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis and y axis</span></span>
<span id="cb55-80"><a href="#cb55-80" aria-hidden="true" tabindex="-1"></a>size_differences_by_treatment_boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> wing_case_size)) <span class="sc">+</span> </span>
<span id="cb55-81"><a href="#cb55-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="co"># Tell ggplot that you want it to build a boxplot</span></span>
<span id="cb55-82"><a href="#cb55-82" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_differences_by_treatment_boxplot) <span class="co"># Print your new figure</span></span>
<span id="cb55-83"><a href="#cb55-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-84"><a href="#cb55-84" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving outputs</span></span>
<span id="cb55-85"><a href="#cb55-85" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;figures/size_differences_by_treatment_boxplot.pdf&quot;</span>, <span class="co"># Give R a path to save to and a file name</span></span>
<span id="cb55-86"><a href="#cb55-86" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> size_differences_by_treatment_boxplot, <span class="co"># Tell R what to save - in this case your object</span></span>
<span id="cb55-87"><a href="#cb55-87" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>) <span class="co"># Tell R what file type to create, in this case a pdf</span></span>
<span id="cb55-88"><a href="#cb55-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-89"><a href="#cb55-89" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a box plot and saving it in an object</span></span>
<span id="cb55-90"><a href="#cb55-90" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis and y axis</span></span>
<span id="cb55-91"><a href="#cb55-91" aria-hidden="true" tabindex="-1"></a>size_differences_by_treatment_and_sex_boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> wing_case_size, <span class="at">fill =</span> sex)) <span class="sc">+</span> </span>
<span id="cb55-92"><a href="#cb55-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="co"># Tell ggplot that you want it to build a boxplot</span></span>
<span id="cb55-93"><a href="#cb55-93" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_differences_by_treatment_and_sex_boxplot) <span class="co"># Print your new figure</span></span>
<span id="cb55-94"><a href="#cb55-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-95"><a href="#cb55-95" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving outputs</span></span>
<span id="cb55-96"><a href="#cb55-96" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;figures/size_differences_by_treatment_and_sex_boxplot.pdf&quot;</span>, <span class="co"># Give R a path to save to and a file name</span></span>
<span id="cb55-97"><a href="#cb55-97" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> size_differences_by_treatment_and_sex_boxplot, <span class="co"># Tell R what to save - in this case your object</span></span>
<span id="cb55-98"><a href="#cb55-98" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>) <span class="co"># Tell R what file type to create, in this case a pdf</span></span>
<span id="cb55-99"><a href="#cb55-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-100"><a href="#cb55-100" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a scatter plot and saving it in an object</span></span>
<span id="cb55-101"><a href="#cb55-101" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis and y axis</span></span>
<span id="cb55-102"><a href="#cb55-102" aria-hidden="true" tabindex="-1"></a>size_vs_parasites_point <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> wing_case_size , <span class="at">y =</span> parasite_burden )) <span class="sc">+</span> </span>
<span id="cb55-103"><a href="#cb55-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="co"># geom_point is ggplots scatter plot</span></span>
<span id="cb55-104"><a href="#cb55-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">&quot;lm&quot;</span>)</span>
<span id="cb55-105"><a href="#cb55-105" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_vs_parasites_point)</span>
<span id="cb55-106"><a href="#cb55-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-107"><a href="#cb55-107" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving outputs</span></span>
<span id="cb55-108"><a href="#cb55-108" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;figures/size_vs_parsite_point.pdf&quot;</span>, <span class="co"># Give R a path to save to and a file name</span></span>
<span id="cb55-109"><a href="#cb55-109" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> size_vs_parasites_point, <span class="co"># Tell R what to save - in this case your object</span></span>
<span id="cb55-110"><a href="#cb55-110" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>) <span class="co"># Tell R what file type to create, in this case a pdf</span></span></code></pre></div>
</details>
<details>
<summary>
<strong>Show me the ggplot fundamentals script</strong>
</summary>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot fundamentals  </span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Ellen Bell </span></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 08/08/2022</span></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up your environment</span></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Reload your tribolium_combo data frame</span></span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a>tribolium_combo <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/tribolium_combo.csv&quot;</span>)</span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Instruct R to treat the treatment variable as a factor </span></span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a>tribolium_combo<span class="sc">$</span>treatment <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(tribolium_combo<span class="sc">$</span>treatment) <span class="co"># Note this will edit your data frame</span></span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a box plot and saving it in an object</span></span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis and y axis</span></span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a>size_differences_boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo, <span class="fu">aes</span>(<span class="at">x =</span> sex, <span class="at">y =</span> wing_case_size)) <span class="sc">+</span> </span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> treatment), <span class="at">width =</span> <span class="fl">0.9</span>) <span class="sc">+</span> <span class="co"># Tell ggplot that you want it to build a boxplot</span></span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Sex&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Wing case size (mm)</span><span class="sc">\n</span><span class="st">&quot;</span>) <span class="sc">+</span> <span class="co"># Adjust your x and y axis labels </span></span>
<span id="cb56-19"><a href="#cb56-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>,<span class="st">&quot;Male&quot;</span>)) <span class="sc">+</span> <span class="co"># Rename the categories on the x axis </span></span>
<span id="cb56-20"><a href="#cb56-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;30°C&quot;</span>, <span class="st">&quot;35°C&quot;</span>), <span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;cornflowerblue&quot;</span>, <span class="st">&quot;coral&quot;</span>)) <span class="sc">+</span> <span class="co"># Rename your treatments</span></span>
<span id="cb56-21"><a href="#cb56-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb56-22"><a href="#cb56-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">color=</span><span class="st">&quot;black&quot;</span>), <span class="co"># Specifies that the plot boarder is coloured black</span></span>
<span id="cb56-23"><a href="#cb56-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes minor grid lines </span></span>
<span id="cb56-24"><a href="#cb56-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes major grid lines </span></span>
<span id="cb56-25"><a href="#cb56-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>), <span class="co"># Changes the size of text on both axis </span></span>
<span id="cb56-26"><a href="#cb56-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>), <span class="co"># Changes size of your axis labels </span></span>
<span id="cb56-27"><a href="#cb56-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>), <span class="co"># Changes the size of text within your legend</span></span>
<span id="cb56-28"><a href="#cb56-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>)) <span class="co"># Changes the size of the legend title  </span></span>
<span id="cb56-29"><a href="#cb56-29" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_differences_boxplot) <span class="co"># Print the object your plot is stored in to view it</span></span></code></pre></div>
</details>
<details>
<summary>
<strong>Show me some examples of scripts for pretty figures</strong>
</summary>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pretty figure examples </span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Ellen Bell </span></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 08/08/2022</span></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up your environment</span></span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb57-7"><a href="#cb57-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-8"><a href="#cb57-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Reload your tribolium_combo data frame</span></span>
<span id="cb57-9"><a href="#cb57-9" aria-hidden="true" tabindex="-1"></a>tribolium_combo <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;data/tribolium_combo.csv&quot;</span>)</span>
<span id="cb57-10"><a href="#cb57-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-11"><a href="#cb57-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Instruct R to treat the treatment variable as a factor </span></span>
<span id="cb57-12"><a href="#cb57-12" aria-hidden="true" tabindex="-1"></a>tribolium_combo<span class="sc">$</span>treatment <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(tribolium_combo<span class="sc">$</span>treatment) <span class="co"># Note this will edit your data frame</span></span>
<span id="cb57-13"><a href="#cb57-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-14"><a href="#cb57-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Make a pretty histogram </span></span>
<span id="cb57-15"><a href="#cb57-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-16"><a href="#cb57-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a histogram and saving it in an object</span></span>
<span id="cb57-17"><a href="#cb57-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis</span></span>
<span id="cb57-18"><a href="#cb57-18" aria-hidden="true" tabindex="-1"></a>size_histogram <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> wing_case_size)) <span class="sc">+</span> </span>
<span id="cb57-19"><a href="#cb57-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">6</span>, <span class="at">fill=</span><span class="st">&quot;cornflowerblue&quot;</span>, <span class="at">color=</span><span class="st">&quot;cornflowerblue&quot;</span>) <span class="sc">+</span> <span class="co"># Tell ggplot that you want it to build a histogram with 6 equal sized bins</span></span>
<span id="cb57-20"><a href="#cb57-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;</span><span class="sc">\n</span><span class="st">Wing Case Size (mm)&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Frequency</span><span class="sc">\n</span><span class="st">&quot;</span>) <span class="sc">+</span> <span class="co"># Adjust your x and y axis labels </span></span>
<span id="cb57-21"><a href="#cb57-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb57-22"><a href="#cb57-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">color=</span><span class="st">&quot;black&quot;</span>), <span class="co"># Specifies that the plot boarder is coloured black</span></span>
<span id="cb57-23"><a href="#cb57-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes minor grid lines </span></span>
<span id="cb57-24"><a href="#cb57-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes major grid lines </span></span>
<span id="cb57-25"><a href="#cb57-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>), <span class="co"># Changes the size of text on both axis </span></span>
<span id="cb57-26"><a href="#cb57-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>), <span class="co"># Changes size of your axis labels </span></span>
<span id="cb57-27"><a href="#cb57-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>), <span class="co"># Changes the size of text within your legend</span></span>
<span id="cb57-28"><a href="#cb57-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>)) <span class="co"># Changes the size of the legend title  </span></span>
<span id="cb57-29"><a href="#cb57-29" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_histogram) <span class="co"># Print your new figure</span></span>
<span id="cb57-30"><a href="#cb57-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-31"><a href="#cb57-31" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;figures/pretty_size_histogram.pdf&quot;</span>, <span class="co"># Give R a path to save to and a file name</span></span>
<span id="cb57-32"><a href="#cb57-32" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> size_histogram, <span class="co"># Tell R what to save - in this case your object</span></span>
<span id="cb57-33"><a href="#cb57-33" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">15</span>, <span class="co"># Set .pdf width</span></span>
<span id="cb57-34"><a href="#cb57-34" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">10</span>, <span class="co"># Set .pdf height</span></span>
<span id="cb57-35"><a href="#cb57-35" aria-hidden="true" tabindex="-1"></a>       <span class="at">units =</span> <span class="st">&quot;cm&quot;</span>, <span class="co"># Specify units for .pdf width and height</span></span>
<span id="cb57-36"><a href="#cb57-36" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>) <span class="co"># Tell R what file type to create, in this case a pdf</span></span>
<span id="cb57-37"><a href="#cb57-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Note, use trial and error to select a good width and height for your figure</span></span>
<span id="cb57-38"><a href="#cb57-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-39"><a href="#cb57-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Make a pretty boxplot</span></span>
<span id="cb57-40"><a href="#cb57-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-41"><a href="#cb57-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a box plot and saving it in an object</span></span>
<span id="cb57-42"><a href="#cb57-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis and y axis</span></span>
<span id="cb57-43"><a href="#cb57-43" aria-hidden="true" tabindex="-1"></a>size_differences_boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo, <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> wing_case_size)) <span class="sc">+</span> </span>
<span id="cb57-44"><a href="#cb57-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> sex), <span class="at">width =</span> <span class="fl">0.9</span>) <span class="sc">+</span> <span class="co"># Tell ggplot that you want it to build a boxplot</span></span>
<span id="cb57-45"><a href="#cb57-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;</span><span class="sc">\n</span><span class="st">Treatment&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Wing case size (mm)</span><span class="sc">\n</span><span class="st">&quot;</span>) <span class="sc">+</span> <span class="co"># Adjust your x and y axis labels </span></span>
<span id="cb57-46"><a href="#cb57-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;30°C&quot;</span>, <span class="st">&quot;35°C&quot;</span>)) <span class="sc">+</span> <span class="co"># Rename the categories on the x axis </span></span>
<span id="cb57-47"><a href="#cb57-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">&quot;Sex&quot;</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>,<span class="st">&quot;Male&quot;</span>), <span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;cornflowerblue&quot;</span>, <span class="st">&quot;coral&quot;</span>)) <span class="sc">+</span> <span class="co"># Rename your treatments</span></span>
<span id="cb57-48"><a href="#cb57-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb57-49"><a href="#cb57-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">color=</span><span class="st">&quot;black&quot;</span>), <span class="co"># Specifies that the plot boarder is coloured black</span></span>
<span id="cb57-50"><a href="#cb57-50" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes minor grid lines </span></span>
<span id="cb57-51"><a href="#cb57-51" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes major grid lines </span></span>
<span id="cb57-52"><a href="#cb57-52" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>), <span class="co"># Changes the size of text on both axis </span></span>
<span id="cb57-53"><a href="#cb57-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>), <span class="co"># Changes size of your axis labels </span></span>
<span id="cb57-54"><a href="#cb57-54" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>), <span class="co"># Changes the size of text within your legend</span></span>
<span id="cb57-55"><a href="#cb57-55" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>)) <span class="co"># Changes the size of the legend title  </span></span>
<span id="cb57-56"><a href="#cb57-56" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_differences_boxplot) <span class="co"># Print the object your plot is stored in to view it</span></span>
<span id="cb57-57"><a href="#cb57-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-58"><a href="#cb57-58" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;figures/pretty_tribolium_wingcase_size_boxplot.pdf&quot;</span>, <span class="co"># Give R a path to save to and a file name</span></span>
<span id="cb57-59"><a href="#cb57-59" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> size_differences_boxplot, <span class="co"># Tell R what to save - in this case your object</span></span>
<span id="cb57-60"><a href="#cb57-60" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">15</span>, <span class="co"># Set .pdf width</span></span>
<span id="cb57-61"><a href="#cb57-61" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">10</span>, <span class="co"># Set .pdf height</span></span>
<span id="cb57-62"><a href="#cb57-62" aria-hidden="true" tabindex="-1"></a>       <span class="at">units =</span> <span class="st">&quot;cm&quot;</span>, <span class="co"># Specify units for .pdf width and height</span></span>
<span id="cb57-63"><a href="#cb57-63" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>) <span class="co"># Tell R what file type to create, in this case a pdf</span></span>
<span id="cb57-64"><a href="#cb57-64" aria-hidden="true" tabindex="-1"></a><span class="co"># Note, use trial and error to select a good width and height for your figure</span></span>
<span id="cb57-65"><a href="#cb57-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-66"><a href="#cb57-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Make a pretty scatter graph</span></span>
<span id="cb57-67"><a href="#cb57-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-68"><a href="#cb57-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Making a scatter plot and saving it in an object</span></span>
<span id="cb57-69"><a href="#cb57-69" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the ggplot function and direct it to your data and define your x axis and y axis</span></span>
<span id="cb57-70"><a href="#cb57-70" aria-hidden="true" tabindex="-1"></a>size_vs_parasites_point <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> tribolium_combo,<span class="fu">aes</span>(<span class="at">x =</span> wing_case_size , <span class="at">y =</span> parasite_burden )) <span class="sc">+</span> </span>
<span id="cb57-71"><a href="#cb57-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">shape =</span> treatment, <span class="at">colour =</span> sex)) <span class="sc">+</span> <span class="co"># geom_point is ggplots scatter plot</span></span>
<span id="cb57-72"><a href="#cb57-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">&quot;lm&quot;</span>, <span class="at">colour =</span> <span class="st">&quot;cornflowerblue&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;lightblue&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb57-73"><a href="#cb57-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;</span><span class="sc">\n</span><span class="st">Wing Case Size (mm)&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Parasite Burden</span><span class="sc">\n</span><span class="st">&quot;</span>) <span class="sc">+</span> <span class="co"># Adjust your x and y axis labels </span></span>
<span id="cb57-74"><a href="#cb57-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">name =</span> <span class="st">&quot;Sex&quot;</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Male&quot;</span>), <span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;deepskyblue3&quot;</span>, <span class="st">&quot;coral&quot;</span>)) <span class="sc">+</span></span>
<span id="cb57-75"><a href="#cb57-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">name =</span> <span class="st">&quot;Treatment&quot;</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;30°C&quot;</span>, <span class="st">&quot;35°C&quot;</span>), <span class="at">values =</span> <span class="fu">c</span>(<span class="dv">15</span>,<span class="dv">16</span>)) <span class="sc">+</span></span>
<span id="cb57-76"><a href="#cb57-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb57-77"><a href="#cb57-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">color=</span><span class="st">&quot;black&quot;</span>), <span class="co"># Specifies that the plot boarder is coloured black</span></span>
<span id="cb57-78"><a href="#cb57-78" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes minor grid lines </span></span>
<span id="cb57-79"><a href="#cb57-79" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="co"># Removes major grid lines </span></span>
<span id="cb57-80"><a href="#cb57-80" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>), <span class="co"># Changes the size of text on both axis </span></span>
<span id="cb57-81"><a href="#cb57-81" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>), <span class="co"># Changes size of your axis labels </span></span>
<span id="cb57-82"><a href="#cb57-82" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>), <span class="co"># Changes the size of text within your legend</span></span>
<span id="cb57-83"><a href="#cb57-83" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>)) <span class="co"># Changes the size of the legend title  </span></span>
<span id="cb57-84"><a href="#cb57-84" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(size_vs_parasites_point)</span>
<span id="cb57-85"><a href="#cb57-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-86"><a href="#cb57-86" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;figures/pretty_tribolium_wingcase_size_parasite_scatter.pdf&quot;</span>, <span class="co"># Give R a path to save to and a file name</span></span>
<span id="cb57-87"><a href="#cb57-87" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> size_vs_parasites_point, <span class="co"># Tell R what to save - in this case your object</span></span>
<span id="cb57-88"><a href="#cb57-88" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">25</span>, <span class="co"># Set .pdf width</span></span>
<span id="cb57-89"><a href="#cb57-89" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">10</span>, <span class="co"># Set .pdf height</span></span>
<span id="cb57-90"><a href="#cb57-90" aria-hidden="true" tabindex="-1"></a>       <span class="at">units =</span> <span class="st">&quot;cm&quot;</span>, <span class="co"># Specify units for .pdf width and height</span></span>
<span id="cb57-91"><a href="#cb57-91" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;pdf&quot;</span>) <span class="co"># Tell R what file type to create, in this case a pdf</span></span>
<span id="cb57-92"><a href="#cb57-92" aria-hidden="true" tabindex="-1"></a><span class="co"># Note, use trial and error to select a good width and height for your figure</span></span></code></pre></div>
</details>
<!--chapter:end:11-merry_christmas.Rmd-->
</div>
<div id="new-year-new-data---workshop-4---athletes-week-2b" class="section level1 hasAnchor" number="13">
<h1 class="hasAnchor"><span class="header-section-number">13</span> New year, new data - Workshop 4 - Athletes, Week 2B<a href="#new-year-new-data---workshop-4---athletes-week-2b" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Good news! There are only three chapters of content remaining. Each chapter from this point onwards is associated with a timetabled 2 hour workshop. However if you don’t complete a chapter in the timetabled slot you are expected to complete it at home. Here we have material for Workshop 4 in Week 2 of Semester 2.</p>
<div id="a-new-project" class="section level2 hasAnchor" number="13.1">
<h2 class="hasAnchor"><span class="header-section-number">13.1</span> A new project<a href="#a-new-project" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will be working on a new data set this term and you will be asked to produce a mock project result section to match the data set. This will help you prepare for the Skills for Biologists portfolio task and act as a good primer for completing other reports that you will be asked to produce throughout your university career. Over the next three workshops we will work through the complete analytical workflow needed to analyse this data set. You can ask for feedback from either myself or demonstrators during your workshop sessions.</p>
<p>In this project you are interested in exploring the differences and relationships in weight, height and red blood cell in a group of male and female athletes.</p>
<div id="project-setup" class="section level3 hasAnchor" number="13.1.1">
<h3 class="hasAnchor"><span class="header-section-number">13.1.1</span> Project setup<a href="#project-setup" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Generally I recommend starting a new R Project every time you start working on new and unrelated data sets. Go to posit Cloud, open a <strong>New R Studio Project</strong> and name it <code>athletes</code>. Spend a few minutes setting up your workspace, refer back to <strong>Chapter 4.3</strong> to remind yourself of how to do this. Remember you will need to create sensible places to save scripts, data and figures. You will also need to freshly install any required packages.</p>
<p>For this workshop series you will need the following packages:</p>
<ul>
<li>tidyverse</li>
<li>patchwork</li>
</ul>
</div>
<div id="script-set-up" class="section level3 hasAnchor" number="13.1.2">
<h3 class="hasAnchor"><span class="header-section-number">13.1.2</span> Script set up<a href="#script-set-up" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Its fairly safe to say we will be creating some new scripts so open a new <strong>R script</strong> and set it up as described in <strong>Chapter 4.5</strong>. The project title will be <em>An analysis of red blood cell counts among athletes</em>. Remember to use <code>library()</code> to load your freshly installed packages.</p>
</div>
<div id="the-data" class="section level3 hasAnchor" number="13.1.3">
<h3 class="hasAnchor"><span class="header-section-number">13.1.3</span> The data<a href="#the-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The data set we will be working on was collected from 202 Australian athletes (102 identified as male (m), 100 identified as female (f)) (data set adapted from Telford and Cunningham, 1991).</p>
<p>The following variables were measured for each athlete:</p>
<ul>
<li>height (ht) was recorded in cm</li>
<li>weight (wt) was recorded in kg</li>
<li>red blood cell count (rcc) in 10<sup>12</sup>l<sup>-1</sup></li>
</ul>
<p>You can download the new data set <a href="https://downgit.github.io/#/home?url=https://github.com/ellenbell/4008Y/blob/main/data/athletes.csv">here</a>. When downloaded you will need to unzip the file, save it somewhere sensible on your computer and then load it in to posit cloud (see <strong>Chapter 4.4</strong> if you are unsure). If you have trouble downloading the data via the link I have also uploaded it to the blackboard Skills for Biologists &gt; Data Sciences &gt; Data page &gt; athletes</p>
</div>
</div>
<div id="check-the-data" class="section level2 hasAnchor" number="13.2">
<h2 class="hasAnchor"><span class="header-section-number">13.2</span> Check the data<a href="#check-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Once you have imported and loaded your athletes.csv spreadsheet into your new r project you will need to spend a few minutes checking it over to make sure that R has correctly identified and formatted your variables.</p>
<div id="task-1---make-sure-your-data-are-clean-and-tidy" class="section level3 hasAnchor" number="13.2.1">
<h3 class="hasAnchor"><span class="header-section-number">13.2.1</span> Task 1 - Make sure your data are clean and tidy<a href="#task-1---make-sure-your-data-are-clean-and-tidy" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>First of all make sure your data have imported correctly, do you have the expected numbers of rows and columns?</li>
</ol>
<blockquote>
<p>Hint - consider using functions such as <code>ncol(), nrow(), colnames()</code> to check this, for more details on how consider revisiting Chapters 4 and 6.</p>
</blockquote>
<ol start="2" style="list-style-type: decimal">
<li>Then check that R has correctly identified the data types for each variable. Do you need to adjust any variables to factors?</li>
</ol>
<blockquote>
<p>Hint - consider using functions such as <code>head()</code> to check and <code>as.factor()</code> to identify variables as factors where required, consider revisiting Chapters 4.4 and 8.1 for further assistance</p>
</blockquote>
<ol start="3" style="list-style-type: decimal">
<li>Now you can check your variable names, are they nice and concise, do you want to change them?</li>
</ol>
<blockquote>
<p>Hint - you can use the <code>rename()</code> function, as seen in Chapter 4.5 to change variable names if you wish to.</p>
</blockquote>
<p>Now we know our that R has read our data correctly and we have all of our variables named as we would like we can run a few further checks. Frequently you will find that your data have originally been manually entered into a spreadsheet, maybe from hand drafted notes in a field or lab notebook. Every time something is copied there is the opportunity for error. Maybe a row has been accidentally duplicated, or there is a typo or maybe some data is missed out altogether. There are a few tricks we can use to check for each of these and to make sure we are confident in the fidelity of our dataset.</p>
<ol style="list-style-type: decimal">
<li>Checking for duplicates - When you are manually entering data into a spreadsheet, it is very easy to accidentally enter the same row twice. With very large data sets this is something that is very hard to pick out by eye. Thankfully R has some very useful functions to check for this. Try running;</li>
</ol>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>athletes <span class="sc">%&gt;%</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">duplicated</span>() <span class="co"># check for duplicated rows</span></span></code></pre></div>
<p><strong>Note that %&gt;% used here is just another notation for piping, rather like the + that ggplot uses. This means that the output from one function is fed directly into the next function.</strong></p>
<p>This chunk of code will spit out a long list of TRUE (row is duplicated) or FALSE (row is not duplicated) statements. Again not very human readable, especially if we have a very large data set. Try ammending the code to;</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>athletes <span class="sc">%&gt;%</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">duplicated</span>() <span class="sc">%&gt;%</span> <span class="co"># check for duplicated rows</span></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sum</span>() <span class="co"># Sums any TRUE statements in the list </span></span></code></pre></div>
<p>Do we have any duplicated rows?</p>
<ol start="2" style="list-style-type: decimal">
<li>Checking for typos - As with duplicates it is very easy to enter a typo when manually entering data into a spreadsheet. Generally if you have been collecting continuous data you will have an idea of what a sensible upper and lower bound within your data set should be. We can use the <code>summarise()</code> function to see what these are within this data set, as shown below;</li>
</ol>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>athletes <span class="sc">%&gt;%</span> </span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">min=</span><span class="fu">min</span>(wt, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="co"># reports the minimum value in the wt variable and ignores cells with N/A</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">max=</span><span class="fu">max</span>(wt, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="co"># reports the maximum value in the wt variable and ignores cells with N/A</span></span></code></pre></div>
<p>Try manipulating the above chunk to report the upper and lower bounds for the height variable in your data set. Do these values all look reasonable to you?</p>
<p>But how can we check for typos in categorical data? We can use the <code>distinct()</code> function to identify all of the options stored under a categorical variable name. Try using the following chunk;</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>athletes <span class="sc">%&gt;%</span> </span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(sex) <span class="co"># reports the categories stored under sex</span></span></code></pre></div>
<p>So do you think there are any potential typos in your data set?</p>
<ol start="3" style="list-style-type: decimal">
<li>Checking for missing data - finally sometimes when entering data manually, you may miss or delete a spreadsheet cell by mistake, leaving it empty. Again this is really difficult to spot by eye in a large data set. Try running the following chunk of code, can you work out what each line does? Try adding comments to it in your script yourself.</li>
</ol>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>athletes <span class="sc">%&gt;%</span> </span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">is.na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sum</span>()</span></code></pre></div>
<details>
<summary>
<strong>Click-me to check your code interpretation</strong>
</summary>
<p>So here you are piping your initial data set <code>athletes</code> into the <code>is.na()</code> function which is then looking for cells containing N/A and reporting a TRUE/FALSE data frame (where TRUE indicates N/A). We are then piping that output straight into the <code>sum()</code> function which is summing the number of TRUE values.</p>
</details>
<p><br />
Do we have any missing data?</p>
</div>
</div>
<div id="explore-the-data" class="section level2 hasAnchor" number="13.3">
<h2 class="hasAnchor"><span class="header-section-number">13.3</span> Explore the data<a href="#explore-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We finished last terms workshops by looking at how we can cannibalise code to fit our purposes. We will be spending the first part of todays workshop going over a few of these skills on our shiney new data set! It may be helpful to look through some of the scripts in <strong>Chapter 12</strong> if you get stuck.</p>
<div id="task-2---look-at-your-data-distributions" class="section level3 hasAnchor" number="13.3.1">
<h3 class="hasAnchor"><span class="header-section-number">13.3.1</span> Task 2 - Look at your data distributions<a href="#task-2---look-at-your-data-distributions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Its always helpful to look at how your data is distributed, as this could have implication for how you tackle downstream analysis.</p>
<p>Create a histogram for the following variables, start with a 6 bins and then make a second set of histogams with 20 bins:</p>
<ul>
<li>Athlete height</li>
<li>Athlete weight</li>
<li>Athlete red blood cell count</li>
</ul>
<blockquote>
<p><strong>Pitt Stop; pause your analysis and check your interpretation</strong>
Compare your six histograms;
How would you describe each plot?
How does changing the bin size effect the shape of the plot?
Look at your histogram for red blood cell count with 20 bins, why do you think it looks like this?</p>
</blockquote>
<details>
<summary>
<strong>Click-me to check your interpretation</strong>
</summary>
<p>Your histograms for height and weight both appear normally distributed. Changing the number of bins doesn’t really alter this. However when you look at the histogram for red blood cell count with 6 bins it looks like it might be skewed, with 20 bins however we get more resolution over the data distribution and two peaks appear. The red blood cell count data are bi-modal. This suggest that we may need to think about our data a little more. These histograms were performed on all the data, which includes both male and female athletes, the bi-modal distribution suggests we should consider splitting the data and analysing male and female athletes separately.</p>
</details>
<p><br />
So lets have a think about how we could split out data set up to examine male and female athletes separately. Try using this chunk;</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pipe your data set to the filter function, this will pull out the male athletes and store then in a new object.</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>male_athletes <span class="ot">&lt;-</span> athletes <span class="sc">%&gt;%</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sex <span class="sc">==</span> <span class="st">&quot;m&quot;</span>)</span></code></pre></div>
<p>Make a similar data frame for female athletes using the above chunk. Then create two new histograms for:</p>
<ul>
<li>Male athlete red blood cell count</li>
<li>Female athlete red blood cell count</li>
</ul>
<blockquote>
<p><strong>Pitt Stop; pause your analysis and check your interpretation</strong>
How would you describe these two new histograms?</p>
</blockquote>
</div>
<div id="task-3---look-for-differences-and-relationships" class="section level3 hasAnchor" number="13.3.2">
<h3 class="hasAnchor"><span class="header-section-number">13.3.2</span> Task 3 - Look for differences and relationships<a href="#task-3---look-for-differences-and-relationships" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now that we have a better idea for how our data are distributed we can start to ask some questions of the data set. First of all think, what kind of plot would you build to explore the following questions? We covered this in <strong>Chapter 7.1</strong>.</p>
<ul>
<li>Is there a difference between male and female athlete height?</li>
<li>Is there a difference between male and female athlete weight?</li>
<li>Is there a difference between male and female athlete red blood cell count?</li>
<li>Is there a relationship between height and weight in male and female athletes?</li>
<li>Is there a relationship between red blood cell count and weight in male and female athletes?</li>
</ul>
<details>
<summary>
<strong>Click-me to check your plot choices</strong>
</summary>
<ul>
<li>Is there a difference between male and female athlete height? - box plot or bar chart</li>
<li>Is there a difference between male and female athlete weight? - box plot or bar chart</li>
<li>Is there a difference between male and female athlete red blood cell count? - box plot or bar chart</li>
<li>Is there a relationship between height and weight in male and female athletes? - scatter plot</li>
<li>Is there a relationship between red blood cell count and weight in male and female athletes? - scatter plot</li>
</ul>
<p>If you are unsure about any of the logic behind these choices - please ask a demonstrator</p>
</details>
<p><br />
Build a draft plot to explore each of your questions.</p>
<p>For your scatter plots add some code to colour the points by sex and add a regression line using <code>geom_smooth()</code>.</p>
<blockquote>
<p><strong>Pitt Stop; pause your analysis and check your interpretation</strong>
How would you interpret each of these plots? Write down a sentence or two to describe the pattern displayed in each. Then look at the scatter plot that explores the relationhsip between red blood cell count and weight in male and female athletes. What does the regression line indicate? Is there a relationship between weight and red blood cell count or could there be another variable effecting the relationship? Remember we know the red blood cell count data are bimodal, how might this effect what we are seeing?</p>
</blockquote>
<details>
<summary>
<strong>Click-me to check your interpretation</strong>
</summary>
<p>Just from glancing at the data, median weight and height are slightly higher in male athletes, this difference is greater when looking at red blood cell counts. There appears to be a positive relationship between weight and height in both males and female athletes, i.e. if we plotted male and female athletes separately, it looks like there would still be a positive relationship between height and weight (we will do this in a moment). When looking at the relationship between red blood cell count and weight in male and female athletes, initially there appears to be a positive relationship, <strong>but</strong> looking at the positioning of male and female points it looks like within each sex the relationship may be less strong. If we think back to our histograms, the red blood cell count data followed a bi-modal distribution, so we need to think about how we explore these data a bit more. In this scatter plot we cant really tell if either of the variables weight or sex explain most of the variation in red blood cell count. So we need to make some more plots to look at the relationship between weight and red blood cell count in male and female athletes separately.</p>
</details>
<p><br />
So it looks like we need to tease our scatter plots apart a little more to make sure our understanding of the data is good. Have a go at making plots for the following four questions (make sure you include a regression line);</p>
<ul>
<li>Is there a relationship between height and weight in female athletes?</li>
<li>Is there a relationship between height and weight in male athletes?</li>
<li>Is there a relationship between red blood cell count and weight in female athletes?</li>
<li>Is there a relationship between red blood cell count and weight in male athletes?</li>
</ul>
<blockquote>
<p><strong>Pitt Stop; pause your analysis and check your interpretation</strong>
Has the relationship changed between weight and height in male and female athletes? Has the relationship changed between red blood cell count and weight in male and female athletes? Do you understand why it was important to explore these scatter graphs a little more? Do you understand why seeing the bi-modal distribution in our histograms was a clue, suggesting that we needed to explore red blood cell count in males and females individually?</p>
</blockquote>
<details>
<summary>
<strong>Click-me to check interpretation</strong>
</summary>
<p>Essentially, the relationship between height and weight in male and female athletes has not changed, and we didn’t really expect it too. By looking at these scatter plots we can see that height explains most of the variation in weight in this data set, makes sense right? Taller people are usually going to be heavier then shorter people. When we look at the relationship between weight and red blood cell count in male and female athletes separately we now see that there is, essentially, no relationship. So weight doesn’t explain the variation in red blood cell count, heavier people don’t necessarily have more red blood cells. So the positive relationship we saw in our first scatter plot was likely explained by sex not weight.</p>
</details>
</div>
</div>
<div id="wrapping-up" class="section level2 hasAnchor" number="13.4">
<h2 class="hasAnchor"><span class="header-section-number">13.4</span> Wrapping up<a href="#wrapping-up" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this chapter we have loaded a new data set into a new R project. We have checked over the data to makes sure that its been loaded correctly and that there are no duplicated rows, typos or missing data. We have then moved on further to exploring the data set, looking at its shape/distribution and identifying any early patterns. As a really important take home message you can hopefully see and appreciate why its important to think about interpretation at each stage of your analysis. If we had not broken our data set down and analysed male and female athletes separately, we may have ended up drawing false conclusions from our data set and this could have derailed much or our downstream analysis as well.</p>
</div>
<div id="before-you-leave-7" class="section level2 hasAnchor" number="13.5">
<h2 class="hasAnchor"><span class="header-section-number">13.5</span> Before you leave!<a href="#before-you-leave-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Log out of posit Cloud and make sure you save your script!</p>
</div>
<div id="references-8" class="section level2 hasAnchor" number="13.6">
<h2 class="hasAnchor"><span class="header-section-number">13.6</span> References<a href="#references-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Pedersen, T. L. (2020). Patchwork: The composer of plots. <a href="https://CRAN.R-project.org/package=patchwork" class="uri">https://CRAN.R-project.org/package=patchwork</a> <br />
Telford, R.D. and Cunningham, R.B. 1991. Sex, sport and body-size dependency of hematology in highly trained athletes. Medicine and Science in Sports and Exercise 23: 788-794. <br />
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. “Welcome to the tidyverse.” Journal of Open Source Software 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686" class="uri">https://doi.org/10.21105/joss.01686</a>. <br />
Wickham, Hadley, Winston Chang, Lionel Henry, Thomas Lin Pedersen, Kohske Takahashi, Claus Wilke, Kara Woo, Hiroaki Yutani, and Dewey Dunnington. 2021. Ggplot2: Create Elegant Data Visualisations Using the Grammar of Graphics. <a href="https://CRAN.R-project.org/package=ggplot2" class="uri">https://CRAN.R-project.org/package=ggplot2</a>.</p>
<!--chapter:end:12-new_year_new_data.Rmd-->
</div>
</div>
<div id="playing-with-some-statistics---workshop-5---athletes-continued-week-4b" class="section level1 hasAnchor" number="14">
<h1 class="hasAnchor"><span class="header-section-number">14</span> Playing with some statistics - Workshop 5 - Athletes continued, Week 4B<a href="#playing-with-some-statistics---workshop-5---athletes-continued-week-4b" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Here we have material for Workshop 5 in Week 4 of Semester 2. We will be continuing with the Athletes data set from Week 2. So go to your <code>athletes</code> project in posit Cloud. It may be helpful to review the plots you made (including the histograms) to remind yourself of how the data look.</p>
<div id="descriptive-statistics" class="section level2 hasAnchor" number="14.1">
<h2 class="hasAnchor"><span class="header-section-number">14.1</span> Descriptive statistics<a href="#descriptive-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>At the end of the Autumn Semester we started to discus some simple descriptive statistics, including measures of central tendency and spread. We discussed means, medians, modes, ranges, standard deviations, standard error of the mean and 95% confidence intervals. If you cant remember what any of these are it is strongly recommended that you revisit the lecture materials. In this section we will be learning how to calculate these basic descriptive statistics for the athletes data set, before moving on to some inferential statistics.</p>
<div id="task-1---calculating-some-descriptive-statistics" class="section level3 hasAnchor" number="14.1.1">
<h3 class="hasAnchor"><span class="header-section-number">14.1.1</span> Task 1 - Calculating some descriptive statistics<a href="#task-1---calculating-some-descriptive-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Copy the following piece of code into your script and then run it;</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a new object called athletes_summary_stats_ht and store mean, standard deviation (sd), and sample size (n) for height (ht) in male and female athletes.</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>athletes_summary_stats_ht <span class="ot">&lt;-</span> athletes <span class="sc">%&gt;%</span> </span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sex) <span class="sc">%&gt;%</span> </span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(ht),</span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">sd=</span><span class="fu">sd</span>(ht),</span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">n=</span><span class="fu">n</span>())</span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a>athletes_summary_stats_ht</span></code></pre></div>
<p>Have a look at the table that R produces. Then look back at the code you used, do you understand what each function has done? The <code>summarise</code> function can also calculate other summary statistics. Try using <code>help(summarise)</code> to find out what else it can do. Play with the function, see if you can produce a similar table that calculates the mean and standard deviation for red blood cell count in male and female athletes.</p>
<p>It would also be useful to know the standard error of the mean and 95% confidence intervals for this data set. We can use the <code>mutate</code> function to add additional columns onto our summary stats table. Remember we can calculate the standard error of the mean using;</p>
<p><span class="math display">\[
SEM = \frac{SD}{\sqrt{n}}
\]</span>
Copy the following chunk of code into your script, run it, and then take a look at the <code>athletes_summary_stats_ht</code> object.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>athletes_summary_stats_ht <span class="ot">&lt;-</span> athletes_summary_stats_ht <span class="sc">%&gt;%</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sem =</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(n))</span></code></pre></div>
<p>Hopefully you should see something that looks like this:</p>
<pre><code>&gt; athletes_summary_stats_ht
# A tibble: 2 × 7
  sex    mean    sd     n   sem 
  &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt;
1 f      175.  8.24   100 0.824
2 m      186.  7.90   102 0.783 </code></pre>
<p>Now try using the <code>mutate</code> function to calculate the 95% confidence intervals. Remember 95% confidence intervals fall on either side of the mean, so you will need an upper and lower bound, so you will need to use mutate twice, once for an <code>upper_ci</code> and once for a <code>lower_ci</code>.</p>
<p>The equation for calculating 95% confidence intervals is:</p>
<p><span class="math display">\[
95percent CI = \ Mean ± (1.96 * SEM)
\]</span></p>
<p>Take a look at your <code>athletes_summary_stats_ht</code> object. It should hopefully look something like this:</p>
<pre><code>&gt; athletes_summary_stats_ht
# A tibble: 2 × 7
  sex    mean    sd     n   sem upper_ci lower_ci
  &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
1 f      175.  8.24   100 0.824     176.     173.
2 m      186.  7.90   102 0.783     187.     184.</code></pre>
<p>The skills you have learned here could be applied to any of the numeric variables in the <code>athletes</code> data set. Try calculating similar values for the red blood cell count variable.</p>
</div>
</div>
<div id="introducing-some-inferential-statistics" class="section level2 hasAnchor" number="14.2">
<h2 class="hasAnchor"><span class="header-section-number">14.2</span> Introducing some inferential statistics<a href="#introducing-some-inferential-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We have, so far, looked at how to explore our data set, both graphically (using histograms, box plots and scatter plots) and numerically (by using descriptive statistics). But something you will come into frequent contact with throughout your degree is inferential statistics. You may heave heard of statistical tests such as; T-test, ANOVA, Chi squared and Mann Whitney U before. These are all different types of statistical model that are loosely grouped under the header of inferential statistics. These are statistics that allow you to make predictions from your data, with the aim of allowing you to take data from your samples and make generalisations about a respective population. You may hear people refer to <em>significant differences</em> or <em>significant relationships</em>, essentially as soon as the term <em>significant</em> comes into play there is an assumption made that some kind of statistical test has been applied.</p>
<p>We wont be going into a huge amount of detail regarding these inferential statistics, the field of statistics is huge and there are many ways you may apply statistics to different data sets. At this stage, I would much prefer that you become confident with visualising data and interpreting those visualisations. However to ensure that you are set up for your future studies and data analysis ambitions, it important that you are at least aware of some inferential statistics. My aim here is go give you an overview of how to run and interpret the statistical models that you are most likely to come across in your degree. This will hopefully act as a foundation in statistic that you can build on as required.</p>
<div id="the-general-linear-model" class="section level3 hasAnchor" number="14.2.1">
<h3 class="hasAnchor"><span class="header-section-number">14.2.1</span> The general linear model<a href="#the-general-linear-model" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>General linear models (not to be confused with generalised linear models or GLMs) are a commonly used statistic in the biological sciences, this is, at least in part, because they are pretty versatile and can output information on statistical significance and effect size. We wont go into the maths behind them in too much detail but in essence they make use of a linear equation. We can apply general linear models to look for differences, if one of our variables is categorical (essentially we are performing an ANOVA or T-test), this will compare the means between our two groups, or we can use the same function to look for relationships and perform a regression, if both of our variables are continuous.</p>
<p>Helpfully, general linear models are very easy to apply in R, there is a function called <code>lm()</code> that is part of the base R package. Lets have a look at some applications of this function.</p>
</div>
<div id="task-2---testing-for-differences" class="section level3 hasAnchor" number="14.2.2">
<h3 class="hasAnchor"><span class="header-section-number">14.2.2</span> Task 2 - Testing for differences<a href="#task-2---testing-for-differences" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Lets say we wanted to analyse the difference in average weight between male and female athletes. So our hypothesis might be; male athletes are heavier then female athletes, in this case the predictor variable is sex and the response variable is weight. If we wanted to use that information to fit a general linear model we could use the following chunk;</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Fitting a linear model </span></span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Here I have created an object &quot;linear_model01&quot; to store the outputs of our model in</span></span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>linear_model01 <span class="ot">&lt;-</span> <span class="fu">lm</span>(wt <span class="sc">~</span> sex, <span class="at">data =</span> athletes) <span class="co"># lm() is the function here, and we are specifying that we want to analyse weight (our response variable) as a function of sex (our predictor variable) using tilde (~). It is super important to place your response and predictor variables on the right side of tilde. We then just tell R which data frame we are using with the data = athletes argument. </span></span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(linear_model01) <span class="co"># Summary() will just print out a summary of our model for us to interpret </span></span></code></pre></div>
<p>The <code>summary()</code> function will then provide you with the following overview of your model output;</p>
<pre><code>&gt; summary(lsmodel01)

Call:
lm(formula = wt ~ sex, data = athletes)

Residuals:
    Min      1Q  Median      3Q     Max 
-29.542  -7.703   0.517   7.538  40.676 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   67.342      1.169  57.599   &lt;2e-16 ***
sexm          15.182      1.645   9.227   &lt;2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 11.69 on 200 degrees of freedom
Multiple R-squared:  0.2986,    Adjusted R-squared:  0.2951 
F-statistic: 85.14 on 1 and 200 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>There is a lot of information here so we will break each section down.</p>
<p>First of all we have a reminder of the formula we gave to the <code>lm()</code> function;</p>
<pre><code>Call:
lm(formula = wt ~ sex, data = athletes)</code></pre>
<p>Then we have a summary of our residuals;</p>
<pre><code>Residuals:
    Min      1Q  Median      3Q     Max 
-29.542  -7.703   0.517   7.538  40.676 </code></pre>
<p>I wont go into a huge amount of detail around how we interpret these, you can read around it if it interests you. But in a nutshell the general linear model draws a straight line through all of our data points, so it has fitted a model predicting where it would expect your data points to land. The residuals are then the distances away from that line each data point is, so the distance each of your observed data points are
from the predicted values. Here R has given us some summary statistics around the residuals.</p>
<p>Next we have our coefficients;</p>
<pre><code>Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   67.342      1.169  57.599   &lt;2e-16 ***
sexm          15.182      1.645   9.227   &lt;2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
</code></pre>
<p>So here we have a table with two rows called <code>(Intercept)</code> and <code>sexm</code> and four collumns called <code>Estimate</code> which is essentially the mean, <code>Std. Error</code> which is standard error, <code>t value</code> which is our T statistic and <code>Pr(&gt;ltl)</code> which is otherwise known as a p value. We set out to model the difference in weight between male and female athletes, but the labeling here is somewhat confusing. The <code>(Intercept)</code> row in this case actually refers to the female athletes, it can help to prove this to yourself, try calculating the mean weight of female athletes and compare the result to the <code>Estimate</code> value. The<code>t value</code> for this row is the <code>Estimate</code> divided by <code>Std. Errors</code> and the <code>Pr(&gt;|t|)</code> is the corresponding value for that t value in a T distribution table, with the given degrees of freedom (don’t worry about this last bit just yet).</p>
<p>A common mistake is therefore to assume that the <code>sexm</code> row then referes to the same values for male athletes. This isn’t quite right. This actually refers to the <strong>difference in the mean weight of the two groups.</strong> So this focuses on the question which we are asking: is their a difference in the weight between male and female athletes? You can check this for yourself, but essentially this row is saying that male athletes are, on average, 15.2kgs heavier then female athletes with a standard error of 1.65kgs. Our <code>t value</code> here is then the <code>Estimate</code> divided by the <code>Std. Error</code> again. This gives our test statistic.</p>
<p>Now lets have a look at our <code>Pr(&gt;ltl)</code> value, otherwise known as a p value. I am not going to go into how these are calculated as they are often worked out by comparing the test statistics (in this case the T value) and degrees of freedom, against a preexisting contingency table. However I will give you some outline of how they are interpreted. Our p value here is given as &gt;2e-16 which translates as 0.0000000000000002. We can interpret this value as there is a 0.0000000000000002% probability that we would see our given t value if the null hypothesis was true. Our null hypothesis being, there is no difference in weight between male and female athletes. This, I hope you will agree, is a very low probability. We generally use cut offs for p values of 0.05 or 0.01, so if you get a p value less then those cut offs you have found statistical significance. Here our p value is less than 0.01 so we can say that there is a statistically significant difference between male and female athlete weight (T-test, T = 9.23, DF = 200, p&lt;0.01).</p>
<p>Finally you may have noticed that there are some asterisks <code>*</code> in the coefficients section, these relate to the <code>Signif.codes</code>, or significance codes row below. Here you can see the relavent number of asterisks and how they relate to the different p value thresholds.</p>
<p>The last part of our summary is shown here;</p>
<pre><code>Residual standard error: 11.69 on 200 degrees of freedom
Multiple R-squared:  0.2986,    Adjusted R-squared:  0.2951 
F-statistic: 85.14 on 1 and 200 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>The residual standard error is pretty much what is says on the tin. I’m not going to go into more depth on residuals here. But I will draw your attention to degrees of freedom or DF. This is essentially the number of independent pieces of information used to calculate a statistic. It’s calculated as the sample size minus the number of restrictions. So in a nutshell its a measure of sample size.</p>
<p>The multiple R-squared value describes how well your regression model explains the variation in your data. Here we have a multiple R-squared of 0.2986, which we could round to 0.3. This can be interpreted as 30% of the variation in weight is explained by the sex of the athletes. Adjusted R-squared, which is also reported here, is only really of use if you are doing multivariate statistics. It takes into account; how many samples you have and how many variables you’re using. Here we only have one variable so we don’t need to worry about the adjusted R-squared.</p>
<p>Finally we have our F statistic. This is essentially another test statistic (like T in the coefficients table), here it has been reported with its own degrees of freedom (DF) and p value. You can interpret these as we did previously.</p>
<p>So by running the <code>lm()</code> function on categorical data and using the <code>summary()</code> function, we have performed both an Independent T-test (t-values and associated p-value in the coefficient section) and an ANOVA (the F statistic and associated p-value). We have also acquired an R-squared value to show the goodness of fit of our regression.</p>
<p>Try running the <code>lm()</code> and <code>summary()</code> functions not to test the null hypothesis that there is no difference in height between male and female athletes, can you interpret the results?</p>
</div>
<div id="task-3---testing-for-relationships" class="section level3 hasAnchor" number="14.2.3">
<h3 class="hasAnchor"><span class="header-section-number">14.2.3</span> Task 3 - Testing for relationships<a href="#task-3---testing-for-relationships" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>When it comes to investigating relationships between two continuous variables we may want to know if there is a statistical correlation between variables. We can use Pearson’s correlation coefficient to do this. Try running the following chunk of code to test for a correlation between height and weight in the athletes data set;</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Correlations</span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>correlation01 <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(athletes<span class="sc">$</span>ht, athletes<span class="sc">$</span>wt, <span class="at">method =</span> <span class="st">&quot;pearson&quot;</span>) <span class="co"># perform a correlation test on height and weight in the athletes data set using Pearson&#39;s correlation coefficient </span></span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>correlation01</span></code></pre></div>
<p>Your output should look somethink like this;</p>
<pre><code>&gt; correlation01

    Pearson&#39;s product-moment correlation

data:  athletes$ht and athletes$wt
t = 17.681, df = 200, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.7205640 0.8295505
sample estimates:
      cor 
0.7809321 
</code></pre>
<p>Here we can see that R has given us a test statistic <code>t</code>, degrees of freedom <code>DF</code> and p-value. R even helpfully tells us that it considers the alternative hypothesis true in this case and that there is a correlation between weight and height variables given that the p-value is less than 0.01. We also have our correlation coefficient labeled here as sample estimates. This is a value between -1 and 1 (-1 being a perfectly negative correlation and 1 being a perfectly positive correlation).</p>
<p>You have probably heard that correlation doesn’t equal causation. But we may wish to see if we can use height to predict weight, to do this we need to try and fit a model to our data. Here we can apply the <code>lm()</code> function again. When we have two continuous variables and use the lm() function, we are performing a linear regression. The linear regression essentially uses the equation for a straight line;</p>
<p><span class="math display">\[
y = mx+c
\]</span></p>
<p>Try running the <code>lm()</code> function to look at the relationship between athlete weight and height (remember to make sure your variables are the correct side of the tilde, you want to analyse weight as a function of height).</p>
<p>Your output should look something like this;</p>
<pre><code>&gt; summary(lsmodel03)

Call:
lm(formula = wt ~ ht, data = athletes)

Residuals:
    Min      1Q  Median      3Q     Max 
-16.372  -5.296  -1.196   4.378  38.031 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -126.19049   11.39566  -11.07   &lt;2e-16 ***
ht             1.11712    0.06318   17.68   &lt;2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 8.72 on 200 degrees of freedom
Multiple R-squared:  0.6099,    Adjusted R-squared:  0.6079 
F-statistic: 312.6 on 1 and 200 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>The output here should look familiar, it is very similar to that produced for the general linear models we fitted before. We can interpret the call and residuals sections of the summary as we did before. However the coefficients section is a little different in its interpretation.</p>
<p>If we look at the <code>Estimate</code> column, the <code>(Intercept)</code> is the intercept in the traditional sense, this would be where the regression line meets the Y axis, when height is zero. Now of course its impossible for someone to weigh -126kgs even if they have a height of 0cms (which is also impossible). But this is simply the fit of the regression line.</p>
<p>Now if you look at the <code>ht</code> <code>Estimate</code> this is 1.12, this is the slope of the regression line. We can also think of this as for every unit increment in height, weight increases by 1.12kg increments as well. We can test this theory.</p>
<p>Plug your intercept and slope values into the y = mx + c equation to estimate the weight of an athlete with a height of 160cms. You can do this by typing <code>1.12 * 160 -126</code> into the console in R. Last week we made scatter plots for for height and weight, with regression lines, have a look and see how our prediction of the weight of an athlete with a height of 160cms compares to the plot.</p>
<p>Regarding the other values represented here, the coefficient <code>Std. Error</code> (Standard Error) measures the average amount that the coefficient estimates vary from the actual average value of our response variable. The <code>t values</code> represent our test statistics and <code>Pr(&gt;|t|)</code> our p values.</p>
<p>The last value I will draw your attention to here is <code>Multiple R-squared</code> value. Here R-squared is 0.61, so rephrased, 61% of the variation in weight can be explained by height. This is quite a large effect size.</p>
</div>
</div>
<div id="assumptions" class="section level2 hasAnchor" number="14.3">
<h2 class="hasAnchor"><span class="header-section-number">14.3</span> Assumptions<a href="#assumptions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We have explored several applications of the general linear model (including T-tests, ANOVA and regression). However you should also be aware of some of the assumptions that these models make. All statistical models come with a set of assumptions that are made of the data set. As a result your choice of test is quite important when starting to perform inferential statistics, and there are a number of factors around your data that you need to be clear about. These include;</p>
<ul>
<li>What is your question and hypothesis?</li>
<li>What types of data are available to you in your data set?</li>
<li>What is your sample size?</li>
<li>What does your data distribution look like?</li>
<li>Is your relationship linear?</li>
<li>Are your observations independent?</li>
<li>We won’t go into this too much this term but you should also consider wether your residuals are normally distributed and homoscedastic (dont worry about this for now).</li>
</ul>
<p>Most of these fairly self explanatory and easy to find out. But a key assumption made by the statistical tests we have looked at so far, and that is a little more tricky to define, is that the data follow a normally distribution.</p>
<p>You will frequently see statistical models refereed to as parametric or non-parametric. Parametric tests assume that the residuals in your data are normally distributed (we can look at the overall distribution of the data as a reasonable proxy for this) and non-parametric tests do not make this assumption, but are not as statistically powerful as parametric tests (hense why we don’t always do a non-parametric test, just to be on the safe side).</p>
<table>
<thead>
<tr class="header">
<th align="center">Parametric</th>
<th align="center">Non-Parametric</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">General linear model variants</td>
<td align="center"></td>
</tr>
<tr class="even">
<td align="center">Independent T-test</td>
<td align="center">Mann-Whitney U</td>
</tr>
<tr class="odd">
<td align="center">ANOVA</td>
<td align="center">Mann-Whitney U</td>
</tr>
<tr class="even">
<td align="center">Pearson’s Correlation</td>
<td align="center">Spearman’s Rank Correlation</td>
</tr>
</tbody>
</table>
<div id="task-4---checking-for-normality" class="section level3 hasAnchor" number="14.3.1">
<h3 class="hasAnchor"><span class="header-section-number">14.3.1</span> Task 4 - Checking for normality<a href="#task-4---checking-for-normality" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To check our distributions we can and should always make a histogram just to look over the data. But there are also tests we can use to reassure ourselves of the data distribution. One such test is called the Shapiro-Wilk test for normality and its very easy to perform in R. Lets go back to our weight variable in the male athletes data set and see if that is normally distributed. First of all pull up your histogram for this variable and remind yourself of how the data are distributed.</p>
<p>Now to add confidence to our interpretation we can run the following piece of code;</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Normality test</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a><span class="fu">shapiro.test</span>(male_athletes<span class="sc">$</span>wt) <span class="co"># Notice we are running this test on just the male athletes</span></span></code></pre></div>
<p>The results should look something like this;</p>
<pre><code>&gt; shapiro.test(athletes$wt)

    Shapiro-Wilk normality test

data:  athletes$wt
W = 0.99299, p-value = 0.4516
</code></pre>
<p>Here our test statistic is W and our p value is fairly self explanatory. But the interpretation of this test is often a little challenging for students. Here the null hypothesis is that our data follow a normal distribution so if our p value is greater then 0.05 we wont reject the null hypothesis and will assume our data are normally distributed. But if our p value is less than 0.05 the distribution of our data is deemed significantly different to that of a normal distribution, so our null hypothesis is rejected and we assume that our data are not normally distributed. In this instance because the weight variable follows a normal distribution it would be fine to run a parametric test on it.</p>
<p>Try running this test on all of your variables for male and female athletes, are any of them not normally distributed?</p>
</div>
<div id="task-5---what-to-do-if-your-data-are-not-normally-distributed" class="section level3 hasAnchor" number="14.3.2">
<h3 class="hasAnchor"><span class="header-section-number">14.3.2</span> Task 5 - What to do if your data are not normally distributed?<a href="#task-5---what-to-do-if-your-data-are-not-normally-distributed" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are a couple of things we can do if our data are not normally distributed.</p>
<ol style="list-style-type: decimal">
<li>We can log transform the data</li>
<li>We can run a non parametric test</li>
</ol>
<p>Hopefully you noticed that the red blood cell count for both male and female athletes are not normally distributed. We can log transform the female althletes red blood cell count using the following piece of code;</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Log transform</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>female_athletes <span class="ot">&lt;-</span> female_athletes <span class="sc">%&gt;%</span></span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">log10 =</span> <span class="fu">log10</span>(female_athletes<span class="sc">$</span>rcc))</span></code></pre></div>
<p>Now try making a new histogram and running the Shapiro-Wilk test on this new log10 variable and see if its normally distributed. Do the same thing for the red blood cell counts in male athletes. How might you interpret these results?</p>
<details>
<summary>
<strong>Click-me to check interpretation</strong>
</summary>
<p>Hopefully you can see that log transforming has reduced the p value for red blood cell counts in female athletes, it is still greater then 0.05, so the data do not follow a normal distribution. The red blood cell count for male athletes following log transformation is normally distributed. But because some of our data does not follow a normal distribution even after a log transformation, we will need to perform a non parametric test.</p>
</details>
<p><br />
The second option with data that aren’t normally distributed is to perform a non-parametric test. If we are interested in looking for differences between categorical variables we can perform a Mann-Whitney U test (instead of a general linear model, ANOVA or T-test) or if we wish to test for correlations we can use Spearmans rank correlation coefficient (instead of Pearsons correlation coefficient).</p>
<p>Try running the following piece of code;</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Mann Whitney U / Wilcoxan </span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(rcc <span class="sc">~</span> sex, <span class="at">data=</span>athletes) </span></code></pre></div>
<pre><code>&gt; wilcox.test(rcc ~ sex, data=athletes) 

    Wilcoxon rank sum test with continuity correction

data:  rcc by sex
W = 948, p-value &lt; 2.2e-16
alternative hypothesis: true location shift is not equal to 0</code></pre>
<p>This is a fairly easy output to interpret. Although this is called a Wilcoxon test, if you run the test on two independent samples (as we have here), R runs a Mann-Whitney U test (confusing I know). Here we have performed a Mann-Whitney U test. The W value is our test statistic and p-value is fairly self explanatory. So we can say that there is a significant difference in red blood cell count between male and female athletes.</p>
<p>Now if we want to investigate a correlation with data that does not follow a normal distribution we can use the Spearmans rank correlation coefficient. Lets use the same functions to test for a correlation between height and red blood cell count. Try using the following chunk;</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Correlations</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>correlation02 <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(athletes<span class="sc">$</span>ht, athletes<span class="sc">$</span>rcc, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>correlation02</span></code></pre></div>
<p>You should get an output that looks something like this;</p>
<pre><code>&gt; correlation02

    Spearman&#39;s rank correlation rho

data:  athletes$ht and athletes$rcc
S = 822518, p-value = 3.262e-09
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.4012392</code></pre>
<p>Here S is the test statistic, the p-value is, well the p-value and our correlation coefficient is represented by sample estimates: rho (this can be interpreted as we did with the Pearson’s test). This again shows that there is a correlation between height and red blood cell count.</p>
</div>
</div>
<div id="wrapping-up-1" class="section level2 hasAnchor" number="14.4">
<h2 class="hasAnchor"><span class="header-section-number">14.4</span> Wrapping up<a href="#wrapping-up-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We have covered a huge amount of ground today, so don’t worry if it takes a while to sink in, or if you need to revisit this chapter a few times. We have covered the basics of how to calculate some descriptive statistics and started playing with performing and interpreting some inferential statistics.</p>
</div>
<div id="before-you-leave-8" class="section level2 hasAnchor" number="14.5">
<h2 class="hasAnchor"><span class="header-section-number">14.5</span> Before you leave!<a href="#before-you-leave-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Log out of posit Cloud and make sure you save your script!</p>
</div>
<div id="references-9" class="section level2 hasAnchor" number="14.6">
<h2 class="hasAnchor"><span class="header-section-number">14.6</span> References<a href="#references-9" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Pedersen, T. L. (2020). Patchwork: The composer of plots. <a href="https://CRAN.R-project.org/package=patchwork" class="uri">https://CRAN.R-project.org/package=patchwork</a> <br />
Telford, R.D. and Cunningham, R.B. 1991. Sex, sport and body-size dependency of hematology in highly trained athletes. Medicine and Science in Sports and Exercise 23: 788-794. <br />
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. “Welcome to the tidyverse.” Journal of Open Source Software 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686" class="uri">https://doi.org/10.21105/joss.01686</a>. <br />
Wickham, Hadley, Winston Chang, Lionel Henry, Thomas Lin Pedersen, Kohske Takahashi, Claus Wilke, Kara Woo, Hiroaki Yutani, and Dewey Dunnington. 2021. Ggplot2: Create Elegant Data Visualisations Using the Grammar of Graphics. <a href="https://CRAN.R-project.org/package=ggplot2" class="uri">https://CRAN.R-project.org/package=ggplot2</a>.</p>
<!--chapter:end:13-playing_with_some_stats.Rmd-->
</div>
</div>
<div id="reporting-your-analysis---workshop-6---athletes-continued-week-6b" class="section level1 hasAnchor" number="15">
<h1 class="hasAnchor"><span class="header-section-number">15</span> Reporting your analysis - Workshop 6 - Athletes continued, Week 6B<a href="#reporting-your-analysis---workshop-6---athletes-continued-week-6b" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This semester we have been working through the complete process of analysing a data set. So we not only do we need to be working on our analytical skills but also thinking about how we report our analyses. Over the last few weeks we have been conducting analysis that will eventually be put into a mock report (to help you prepare for future reports that you may need to produce). So lets begin by thinking about how we might present some of our figures.</p>
<div id="selecting-what-to-report" class="section level2 hasAnchor" number="15.1">
<h2 class="hasAnchor"><span class="header-section-number">15.1</span> Selecting what to report<a href="#selecting-what-to-report" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>When you are conducting your analysis you will produce many plots, however you will not necessarily need to present all of these plots. Some analyses you perform as part of your data exploration and will not be used in the final write up. Being good at selecting what to report is quite an undervalued skill.</p>
<div id="task-1---start-thinking-about-how-you-might-report-your-data" class="section level3 hasAnchor" number="15.1.1">
<h3 class="hasAnchor"><span class="header-section-number">15.1.1</span> Task 1 - Start thinking about how you might report your data<a href="#task-1---start-thinking-about-how-you-might-report-your-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First of all, identify which plots you would include in a report, if you were interested in answering the following questions;</p>
<ul>
<li>Is there a difference between male and female athlete height?</li>
<li>Is there a difference between male and female athlete weight?</li>
<li>Is there a difference between male and female athlete red blood cell count?</li>
<li>Is there a relationship between height and weight in male and female athletes?</li>
<li>Is there a relationship between red blood cell count and weight in male and female athletes?</li>
</ul>
<blockquote>
<p><strong>Selecting plots for a report</strong>
An extremely common error students make is to include every plot they build in their report. Sometimes it is important to build a plot, as part of your own data exploration, but not include it when writing your report. Histograms are a prime example of this, unless the question you are exploring is related to data distribution, you do not need to include histograms in your project reports. You do need to know how the data are distributed, but you do not need to present the histograms that show the data distribution. Your report should only include figures that relate to the questions you are asking. You want to try and tell a story through your figures, while showing academic rigour. With this in mind, which of your plots would you include in your report, given the questions above?</p>
</blockquote>
<details>
<summary>
<strong>Click-me to see which figures Ellen would present</strong>
</summary>
<p>There are several combinations of plots you could present to answer the above questions in a report, but this is how I would do it;</p>
<ul>
<li>Figure 1 - A multi-panel (group of plots) figure containing box plots for (i) weight in male and female athletes, (ii) height in male and female athletes and (iii) scatter plot for weight vs height in male and female athletes (pooled)</li>
<li>Figure 2 - a multi-panel figure containing (i) box plot for red blood cell count in male and female athletes, (ii) scatter plot for red blood cell counts in male athletes, (iii) scatter plot for red blood cell counts in female athletes</li>
</ul>
We will cover how to make multi-panel figures shortly.
</details>
<p><br />
Make each figure that you would include in your report visually pleasing. Use your ggplot skills and scripts to do this. Think about colour choices and try to make them uniform.</p>
</div>
<div id="task-2---building-your-first-multi-pannel-figure" class="section level3 hasAnchor" number="15.1.2">
<h3 class="hasAnchor"><span class="header-section-number">15.1.2</span> Task 2 - Building your first multi-pannel figure<a href="#task-2---building-your-first-multi-pannel-figure" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Above I suggested that it may be a good idea to present some of your reported plots in a multi-panel figure. We can make these in R using the package <code>patchwork</code>. See if you can install and load the <code>patchwork</code> package and then take a look at the the chunk below for a demonstration of how it can be used;</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Demonstrating how to use the plot_layout function within the patchwork package to make multi-panel figures</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="co"># First, here are some basic plots, you can use your own plots that have been nicely formatted instead, these are just to demonstrate</span></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>weight_box <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> athletes,<span class="fu">aes</span>(<span class="at">x =</span> sex, <span class="at">y =</span> wt)) <span class="sc">+</span> </span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() </span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>height_box <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> athletes,<span class="fu">aes</span>(<span class="at">x =</span> sex, <span class="at">y =</span> ht)) <span class="sc">+</span> </span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() </span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>height_v_weight <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> athletes,<span class="fu">aes</span>(<span class="at">x =</span> ht, <span class="at">y =</span> wt)) <span class="sc">+</span> </span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> sex)) <span class="sc">+</span></span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">&quot;lm&quot;</span>)</span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Now we can use the plot_layout function within the patchwork package to make our multi-panel figure.</span></span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Here I have asked for the height and weight box plots to be on the top row, side by side and the height v weight scatter below.</span></span>
<span id="cb84-13"><a href="#cb84-13" aria-hidden="true" tabindex="-1"></a><span class="co"># You can play with the configuration to suit your tastes, use help(plot_layout) to see some other configurations.</span></span>
<span id="cb84-14"><a href="#cb84-14" aria-hidden="true" tabindex="-1"></a><span class="co"># You may notice that here I have specified the package for the function using &quot;::&quot; e.g. patchwork::plot_layout.</span></span>
<span id="cb84-15"><a href="#cb84-15" aria-hidden="true" tabindex="-1"></a><span class="co"># This is because some functions across multiple packages have shared names, as the do here, so R can become confused, by specifying the package, I am directing R to the specific package which contains the specific function I wish to use. </span></span>
<span id="cb84-16"><a href="#cb84-16" aria-hidden="true" tabindex="-1"></a><span class="co"># The final argument (guides=&#39;collect&#39;) pulls together all the little figure keys you may have and presents them on one place and should remove duplicated information</span></span>
<span id="cb84-17"><a href="#cb84-17" aria-hidden="true" tabindex="-1"></a>(height_box <span class="sc">|</span> weight_box) <span class="sc">/</span> height_v_weight <span class="sc">+</span> patchwork<span class="sc">::</span><span class="fu">plot_layout</span>(<span class="at">guides =</span><span class="st">&#39;collect&#39;</span>)</span></code></pre></div>
<p>The results of this chunk can be seen in Figure 13.1. Note that these are rough draft figures and placement, you can play with the ggplot and patchwork parameters to get them to look as you would like, and as would be expected in a written report.</p>
<div class="figure">
<img src="figures/screenshot_patchwork.png" alt="Results from the patchwork demonstration chunk" width="90%" />
<p class="caption">
(#fig:unnamed-chunk-77)Results from the patchwork demonstration chunk
</p>
</div>
<p>Have a go at putting some of your report ready plots into a multi-panel figure. Use <code>ggsave</code> to save all of the figures (multi-panel or single-panel). Check your pdf sizes to make sure they are scaled sensibly, you can change the scaling options by looking at the code chunks in <strong>Chapter 12</strong>. You can download these pdfs by opening them in browser and clicking the download button in the top right hand corner.</p>
</div>
</div>
<div id="writing-about-your-analysis" class="section level2 hasAnchor" number="15.2">
<h2 class="hasAnchor"><span class="header-section-number">15.2</span> Writing about your analysis<a href="#writing-about-your-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We have done a lot of exploration when it comes to presenting data and performing analysis. But something you will frequently be asked to do throughout your degree, is write about your findings. Here I will cover a few summary point to bare in mind when writing about your analyses.</p>
<ul>
<li>Keep it concise - always aim to keep sentences short and snappy when writing in the sciences</li>
<li>Be specific - make sure it is very clear what it is you are discussing, especially when you have a project with multiple variables</li>
<li>Keep it biology focused - statistics, be they inferential or descriptive, are helpful when building an argument, you can use them to support your point, but the point should be biology driven.</li>
<li><strong>Always</strong> give a measure of spread, like standard error, when discussing a mean.</li>
</ul>
<p><strong>Do</strong> this; <br /></p>
<p>“Male gorillas ate fewer bananas per day (mean ± s.e. = 12.3 ± 1.3) compared to females (mean ± s.e. = 14.7 ± 1.1)” <br /></p>
<p><strong>Do NOT</strong> do this; <br /></p>
<p>“The mean was 12.3 for males and 14.7 for females so females ate more bananas.” <br /></p>
<p>This second example makes the entire sentence about the mean values and not about the biological question, which is the difference in bananas eaten by male and female gorillas. It also lacks details like standard deviation or error but discusses a mean which by itself is not very meaningful. The first example was a much more professional and scientifically sound way to report this analysis, using values from the analysis to support the main argument. It will take practice to introduce this writing method into your own work. <br /></p>
<p>You may have noticed that in papers and published reports statistics are reported in a very specific way. When reporting statistics, you should try to state what is shown and then include supporting information, in brackets, from the statistical analysis you have run. When reporting statitics, you should report;</p>
<ul>
<li>The statistical test used (e.g. T-test, ANOVA, etc)</li>
<li>The test statistic (T value or F value for T-tests or ANOVA)</li>
<li>Degrees of freedom or some alternative indicator of sample size as recomended by the test run</li>
<li>p value</li>
</ul>
<p>For example, <br /></p>
<p><strong>Do</strong> this; <br /></p>
<p>“Male gorillas ate fewer bananas per day (mean ± s.e. = 12.3 ± 1.3) compared to females (mean ± s.e. = 14.7 ± 1.1) and this difference was significant (T-test, t = -3.7, d.f. = 99, P = 0.035)”</p>
<p><strong>Do NOT</strong> do this; <br /></p>
<p>“The mean was 12.3 for males and 14.7 for females so females ate more bananas. The test was significant because p&lt; 0.05” <br /></p>
<p>This second example is very commonly seen in student work and should be avoided at all costs. The analyst here is asking the reader to accept that there was a statistical difference because p was less than an the arbitrary cut off of 0.05. It gives no evidence to show that this is true. It also creates the impression that the person who conducted the analysis had no understanding of how or why they were analysing the data or what it was they were looking for. The first example is only fractionally longer, but provides much more information to the reader, it is professional and allows the reader to feel confident that the analyst is basing their conclusion on sound scientific and statistical evidence.</p>
<div id="task-3---writing-a-figure-legend" class="section level3 hasAnchor" number="15.2.1">
<h3 class="hasAnchor"><span class="header-section-number">15.2.1</span> Task 3 - Writing a figure legend<a href="#task-3---writing-a-figure-legend" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You will often need to construct figure legends, here are a few pointers to bare in mind;</p>
<ul>
<li>Table legends go above and figure legends go below</li>
</ul>
<p>Legends <strong>should</strong>;</p>
<ul>
<li>Allow the reader to understand the figure without reference to the text, so include information about what is being shown. For example, with a box plot you may wish to say “The middle line, box and whiskers represent the median, interquartile range and range, respectively.”</li>
<li>Include info on sample size, study organism (in Latin) and (if relevant, e.g. ecology) study location and time</li>
</ul>
<p>Legends <strong>should NOT</strong>;</p>
<ul>
<li>Include information that is obvious to a scientific audience (e.g. “a scatter plot to show” or “a graph to show”)</li>
<li>Repeat what’s in the figure (e.g. describe what the colour mean if you also have a key in the figure)</li>
</ul>
<p>With this in mind have a go at writing a figure legend for some of the report ready figures that you made earlier.</p>
</div>
<div id="task-4---have-a-go-at-preparing-a-result-section" class="section level3 hasAnchor" number="15.2.2">
<h3 class="hasAnchor"><span class="header-section-number">15.2.2</span> Task 4 - Have a go at preparing a result section<a href="#task-4---have-a-go-at-preparing-a-result-section" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You have some report ready figures and some report ready figure legends now. The next step is to put this together into a result section. Have a think about how you would describe the data you are presenting. It only needs to be a couple of paragraphs. Have a demonstrator check your work.</p>
</div>
</div>
<div id="wrapping-up-2" class="section level2 hasAnchor" number="15.3">
<h2 class="hasAnchor"><span class="header-section-number">15.3</span> Wrapping up<a href="#wrapping-up-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This is the final content driven Data Sciences workshop. This term we have been slowly progressing through the complete workflow of the analysis of a small data set. We started by importing, checking and cleaning our data, we then did some data exploration and made some exploratory plots before using descriptive and inferential statistics to investigate differences and relationships in more depth. Finally, today, we have looked at the processes of selecting and preparing plots for presentation and how to write about our findings. The next chapter contains the Data Sciences portion of the Skills for Biologists portfolio task, which is the summative assignment for this term. There is one more workshop which is a drop in session, you can get help on your code at this session, but we cannot help with other aspects of your assignment.</p>
</div>
<div id="before-you-leave-9" class="section level2 hasAnchor" number="15.4">
<h2 class="hasAnchor"><span class="header-section-number">15.4</span> Before you leave!<a href="#before-you-leave-9" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Log out of posit Cloud and make sure you save your script!</p>
</div>
<div id="references-10" class="section level2 hasAnchor" number="15.5">
<h2 class="hasAnchor"><span class="header-section-number">15.5</span> References<a href="#references-10" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Pedersen, T. L. (2020). Patchwork: The composer of plots. <a href="https://CRAN.R-project.org/package=patchwork" class="uri">https://CRAN.R-project.org/package=patchwork</a>
Telford, R.D. and Cunningham, R.B. 1991. Sex, sport and body-size dependency of hematology in highly trained athletes. Medicine and Science in Sports and Exercise 23: 788-794. Pedersen, T. L. (2020). Patchwork: The composer of plots. <a href="https://CRAN.R-project.org/package=patchwork" class="uri">https://CRAN.R-project.org/package=patchwork</a> <br />
Telford, R.D. and Cunningham, R.B. 1991. Sex, sport and body-size dependency of hematology in highly trained athletes. Medicine and Science in Sports and Exercise 23: 788-794. <br />
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. “Welcome to the tidyverse.” Journal of Open Source Software 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686" class="uri">https://doi.org/10.21105/joss.01686</a>. <br />
Wickham, Hadley, Winston Chang, Lionel Henry, Thomas Lin Pedersen, Kohske Takahashi, Claus Wilke, Kara Woo, Hiroaki Yutani, and Dewey Dunnington. 2021. Ggplot2: Create Elegant Data Visualisations Using the Grammar of Graphics. <a href="https://CRAN.R-project.org/package=ggplot2" class="uri">https://CRAN.R-project.org/package=ggplot2</a>.</p>
<!--chapter:end:14-reporting_analysis.Rmd-->
</div>
</div>
<div id="portfolio-tasks" class="section level1 hasAnchor" number="16">
<h1 class="hasAnchor"><span class="header-section-number">16</span> Portfolio Tasks<a href="#portfolio-tasks" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="background" class="section level2 hasAnchor" number="16.1">
<h2 class="hasAnchor"><span class="header-section-number">16.1</span> Background<a href="#background" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The data you will be analysing focus’ on parasite abundance in <em>Corydoras</em> catfish.</p>
<div class="figure">
<img src="figures/cory.jpg" alt="*Corydoras araguaiaensis*" width="90%" />
<p class="caption">
(#fig:unnamed-chunk-78)<em>Corydoras araguaiaensis</em>
</p>
</div>
<p><em>Corydoras</em> catfishes were collected from Brazil and identified. Each fish belonged to one of two species that are very similar in life history but have dramatically different genome sizes. <em>Corydoras araguaiaensis</em> is a polyploid species (probable tetraploid), and <em>Corydoras maculifer</em> is a diploid species. The two species are müllerian mimics and coexist, sharing a niche, so environmentally they have had very similar experiences.</p>
<p>This prediction was that polyploid fish may carry a higher number of different immune alleles and therefore may be better able to resist parasites than the diploid species. The fish were dissected and number of parasites found within each individual was counted.</p>
</div>
<div id="the-data-1" class="section level2 hasAnchor" number="16.2">
<h2 class="hasAnchor"><span class="header-section-number">16.2</span> The data<a href="#the-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The data you have been provided with can be found <a href="https://downgit.github.io/#/home?url=https://github.com/ellenbell/4008Y/blob/main/data/catfish_parasite_data.csv">here</a>, it is also on blackboard should the download link prove troublesome (blackboard Skills for Biologists &gt; Data Sciences &gt; Data page &gt; catfish_parasite_data).</p>
<p>The data consist of four variables;</p>
<ol style="list-style-type: decimal">
<li>sample - this is simply a sample ID with values ranging from 1-61</li>
<li>length_mm - standard length in millimetres. Standard length is a method of measurement for fish and is recorded as the distance from the tip of the snout to the base of the tail.</li>
<li>species - either the polyploid <em>C. araguaiaensis</em> (A) or the diploid <em>C. maculifer</em> (M)</li>
<li>parasite_count - a measure of total parasites counted per individual</li>
</ol>
<p>This data set was adapted from Bell et al., 2020.</p>
</div>
<div id="the-tasks" class="section level2 hasAnchor" number="16.3">
<h2 class="hasAnchor"><span class="header-section-number">16.3</span> The tasks<a href="#the-tasks" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Task 3 and 4 of your Skills portfolio are based in the Data Sciences work you have been doing. Each of these is broken into two parts - each exercise listed below describes the work required to complete each part of the two Data Sciences portfolio tasks. If you wish to know how these sections will be marked, consult the Skills for Biologists Portfolio Rubric on Blackboard.</p>
<div id="exercise-1---plots" class="section level3 hasAnchor" number="16.3.1">
<h3 class="hasAnchor"><span class="header-section-number">16.3.1</span> Exercise 1 - plots<a href="#exercise-1---plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You wish to know if there is a link between genome size and parasite load in <em>Corydoras</em> catfishes, but you also want to make sure and show that there are no confounding variables (like size) affecting any conclusions you draw from your analysis. So you ask three questions of your data;</p>
<ol style="list-style-type: decimal">
<li>Is there a difference in length between <em>C. araguaiaensis</em> and <em>C. maculifer</em>?</li>
<li>Is there a difference in parasite abundance between <em>C. araguaiaensis</em> and <em>C. maculifer</em>?</li>
<li>Is there a relationship between length and parasite abundance in <em>C. araguaiaensis</em> and <em>C. maculifer</em>?</li>
</ol>
<p>Set up a new R studio project, import the catfish_parasites data and make sure it is clean and tidy. Complete all the checks that you did in <strong>Chapter 13</strong> and then you can start exploring the data (<strong>Chapter 13 and 14</strong>). When you are satisfied with the data quality and appearance you can begin to answer the questions above. Construct a plot to address each of these questions. I suggest a plot per question. Make them visually pleasing and then combine the three plots into a multi-panel figure (<strong>Chapter 15</strong>). Export this as a pdf, you will be able to download this pdf to your own computer when you are happy with it (you may need to open it in your browser and then click the download button in the top right hand corner of the screen). Present these plots in your portfolio for Task 3.</p>
</div>
<div id="exercise-2---figure-legend" class="section level3 hasAnchor" number="16.3.2">
<h3 class="hasAnchor"><span class="header-section-number">16.3.2</span> Exercise 2 - figure legend<a href="#exercise-2---figure-legend" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We have discussed figure legends in lectures and in <strong>Chapter 15</strong>, write a good figure legend underneath your newly made figure within Task 3 of your portfolio.</p>
</div>
<div id="exercise-3---justification" class="section level3 hasAnchor" number="16.3.3">
<h3 class="hasAnchor"><span class="header-section-number">16.3.3</span> Exercise 3 - justification<a href="#exercise-3---justification" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Task 4 of your portfolio is divided into two parts, a justification and a proof of programming. The first part is a justification of your work flow. Please briefly bullet point a summary of the process you went through to complete Exercise 1 and why you made the choices you made, this should extend from the point that you downloaded the data set to presenting your final figure.</p>
</div>
<div id="exercise-4---proof-of-programing" class="section level3 hasAnchor" number="16.3.4">
<h3 class="hasAnchor"><span class="header-section-number">16.3.4</span> Exercise 4 - proof of programing<a href="#exercise-4---proof-of-programing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The second part of Task 4 is a proof of programming. Please make a screen grab of one of the chunks of code used to create one of your plots. Make sure we can clearly see the code (an example of a good printscreen is shown in Figure 16.2).</p>
<div class="figure">
<img src="figures/code_example.png" alt="ggplot chunk" width="90%" />
<p class="caption">
(#fig:unnamed-chunk-79)ggplot chunk
</p>
</div>
<p>Good luck everyone! The final workshop in Week 8 will be a drop in session, so if you are really stuck do feel free to pop by. We can help you with fixing code but cant give you more feedback then that.</p>
</div>
</div>
<div id="references-11" class="section level2 hasAnchor" number="16.4">
<h2 class="hasAnchor"><span class="header-section-number">16.4</span> References<a href="#references-11" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Bell, E., Cable, J., Oliveira, C., Richardson, D., Yant, L. &amp; Taylor, M. (2020); Help or Hindrance? The evolutionary impact of whole-genome duplication on immunogenetic diversity and parasite load. Ecology and Evolution. 10 (24). 13949-13956.</p>
<!--chapter:end:15-portfolio_task.Rmd-->
</div>
</div>
<!--bookdown:body:end-->
            </section>

          </div>
        </div>
      </div>
<!--bookdown:link_prev-->
<!--bookdown:link_next-->
    </div>
  </div>
<!--bookdown:config-->

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
